% Use only LaTeX2e, calling the article.cls class and 12-point type.

\documentclass[11pt]{article}

\usepackage[round,semicolon]{natbib}
\usepackage{etoolbox}
\AtBeginEnvironment{quote}{\singlespacing\tiny}
% Use times if you have the font installed; otherwise, comment out the
% following line.

% added by SKH
%\usepackage{lineno}
%\linenumbers

\usepackage{times}
\usepackage{amssymb}
\usepackage{amsmath}

\usepackage[export]{adjustbox}

\usepackage{graphicx}
\graphicspath{ {images/} }

% for adjustwidth
\usepackage{changepage}

% The following parameters seem to provide a reasonable page setup.

\topmargin 0.0cm
\oddsidemargin 0.2cm
\textwidth 16cm 
\textheight 21cm
\footskip 1.0cm

\usepackage{newfloat}
\usepackage{amsmath}
\usepackage[labelfont=bf]{caption}
\usepackage{nameref}
\usepackage{rotating}
\usepackage{color}
\usepackage{float}
\renewcommand{\figurename}{{}}
\renewcommand{\thefigure}{{Figure \arabic{figure}}}

\renewcommand{\tablename}{{}}
\renewcommand{\thetable}{{Table \arabic{table}}}

\newfloat{suppfile}{thp}{losuppfile}
\renewcommand{\thesuppfile}{Supplementary file \arabic{suppfile}}
\floatname{suppfile}{}

\newfloat{suppfig}{thp}{losuppfig}
\renewcommand{\thesuppfig}{Supplementary figure \arabic{suppfig}}
\floatname{suppfig}{}

%
\newfloat{supptable}{thp}{losupptable}
\renewcommand{\thesupptable}{Supplementary table \arabic{supptable}}
\floatname{supptable}{}
%

\renewcommand{\theequation}{Equation \arabic{equation}}

\newcommand{\mutDNA}{\textbf{mutDNA}}
\newcommand{\mutvirus}{\textbf{mutvirus}}
\newcommand{\DNA}{\textbf{DNA}}
\newcommand{\virus}{\textbf{virus}}

\newcommand\comment[1]{{\color{magenta}#1}}


\usepackage{hyperref}
\hypersetup{colorlinks,citecolor=blue,linkcolor=blue,urlcolor=blue}
\hypersetup{colorlinks,citecolor=blue,linkcolor=blue,urlcolor=blue}

\usepackage{seqsplit}

\usepackage{array}
\newcolumntype{P}[1]{>{\raggedright\arraybackslash}p{#1}}

\title{Experimentally Informed Site-Specific Substitution Models Substantially Deepen Divergence Estimates} 

\author
{Sarah K. Hilton$^{1,2}$  and Jesse D. Bloom$^{1,2,}$\\
\\
\normalsize{$^1$Division of Basic Sciences and Computational Biology Program,}\\
\normalsize{Fred Hutchinson Cancer Research Center, Seattle, WA 98109, USA}\\
\normalsize{$^2$Department of Genome Sciences, University of Washington, Seattle, WA}\\
\normalsize{E-mail:  jbloom@fredhutch.org.}\\
}


% Include the date command, but leave its argument blank.

\date{}

\usepackage{setspace}
\onehalfspacing


\begin{document} 

% Double-space the manuscript.


% Make the title.

\maketitle 


\begin{abstract}
\textit{$\leq$ 250 words, currently 159}   

Molecular dating techniques have been used to estimate the divergence date of many viruses. 
However, these estimates are consistently substantially younger than estimates from methods which are independent of branch length estimation. 
This discrepancy is caused, in part, by inadequate modeling of purifying selection leading to branch length underestimation. 
Here, we show that substitution models informed by empirical measurements of mutational constraint better than traditional models and extend branch lengths. 
We used models informed by deep mutational scanning experiments performed in two, highly diverged influenza virus hemagglutinin homologs to optimize the branch lengths of a phylogenetic tree. 
For each experimentally informed model, we observed extension in branch length from the experiment's focal sequence. 
This extension in branch length due to explicit modeling of site-specific purifying selection is observed in the presence and absence of standard methods for modeling site-to-site variation. 
Overall, this study underscores the importance of modeling purifying selection when estimating branch lengths and, by extension, divergence dates. 

\end{abstract}

\clearpage

\section*{Introduction} 

\textit{Introduction outline}    
\textit{1. Estimating the divergence date of viruses is important and common.}  
\textit{2. There are numerous examples of molecular dating techniques contradicting other dating methods for viruses.}  
\textit{3. Why purifying selection could cause branch length underestimation.}  
\textit{4. Discussion of the attempts/methods made thus far to model site-to-site rate variation.}  
\textit{5. We use empirical measurements to model site-specific purifying selection. }  
\textit{6. In this study we compare branch lengths of trees optimized by GY94 and ExpCM models and see an extension in branch lengths with ExCM and with $\Gamma\omega$. }  

\textbf{Estimating the age of viruses is important to understanding their evolutionary history, including co-evolution with their host species.}
Molecular dating techniques are commonly used to estimate the dates of viruses. 
However, these dates are consistently younger than the dates obtained by other methods which are independent of phylogenetic trees, often by orders of magnitude. 
\textit{example HIV/SIV, foamy virus, or measles}

\textbf{Failing to account for site-specific purifying selection results in an underestimation of branch lengths.} 
It has long been observed that protein-coding genes have a non-uniform distribution of amino-acid frequencies across the sites in the protein. 
Multiple sequence alignments of homologs show some sites which are conserved and appeared to be mutationally constrained and other sites which are variable and appear to be mutationally tolerant. 
This site-specific purifying selection dictates the expectation of change at a given site which is translated into branch length. 
\textit{The strength of purifying selection at a given site gives the expectation of change which directly relates to branch length. 
Specifically, failing to account for constraint will lead to an underestimation in branch length because your expectation of change will be too high.} 

\textbf{Different substitution models have been developed to address the effect of site-specific purifying selection.}
Most commonly, a rate parameter is described by some statistical distribution fit across the entire protein, such as the $\Gamma$-distributed $\omega$ in the YNGKP M5 \citep{yang2000codon}. 
Other approaches have tried to explicitly model the site-specific amino-acid frequencies, such as mutation-selection models \citep{halpern1998evolutionary}. 
While perhaps the most biologically-relevant, mutation-selection models are heavily parametrized and run the risk of overfitting in most practical applications. 
However, we have previously shown that we can model this site-specific purifying selection using empirical measurements of mutational constraint from a high-throughput assay called deep mutational scanning. 
We hypothesize that using these experimentally informed models will estimate longer branches than traditional models. 

\textbf{Here, we address this hypothesis by comparing the branch lengths on phylogenetic trees describing influenza virus hemagglutinin optimized by different substitution models.}
By comparing the trees optimized with and without the experimental descriptions of purifying selection, we show that failing to account for site-specific constraints results in shorter estimations of branch length. 
We used two different experimentally informed models, defined by experimental measurements from one of two diverged homologs. 
For each model, the branchs from the experimental focal sequence are extended compared to traditional models.  
This branch length extension is seen even in the presence of model parameters which are traditionally used to describe site-to-site rate variation.  
These results underscore the importance of modeling purifying selection when estimating divergence dates and suggest that experimental measurement may be used to account for this selection while avoiding overparameterization. 

\section*{Results and Discussion}

\subsection*{Substitution models}
\subsubsection*{GY94 models}
\subsubsection*{ExpCMs}
We recap the \textbf{Exp}erimentally Informed \textbf{C}odon \textbf{M}odel (ExpCM) \citep{bloom2014experimentally,bloom2014informed,bloom2017identification,hilton2017phydms} to introduce nomenclature. 

In an ExpCM, rate of substitution $P_{r,xy}$ of site $r$ from codon $x$ to $y$ is written in mutation-selection form~\citep{halpern1998evolutionary,mccandlish2014modeling,spielman2015relationship} as
\begin{equation}
P_{r,xy} = Q_{xy} \times F_{r,xy}
\end{equation}
where $Q_{xy}$ is proportional to the rate of mutation from $x$ to $y$, and $F_{r,xy}$ is proportional to the probability that this mutation fixes.
The rate of mutation $Q_{xy}$ is assumed to be uniform across sites, and takes an HKY85-like~\citep{hasegawa1985dating} form:
\begin{equation}
Q_{xy} = 
\begin{cases}
\phi_w & \mbox{if $x$ and $y$ differ by a transversion to nucleotide $w$} \\
\kappa \phi_w & \mbox{if $x$ and $y$ differ by a transition to nucleotide $w$} \\
0 & \mbox{if $x$ and $y$ differ by $>1$ nucleotide.}
\end{cases}
\end{equation}
The $\kappa$ parameter represents the transition-transversion ratio, and the $\phi_w$ values give the expected frequency of nucleotide $w$ in the absence of selection on amino-acid substitutions, and are constrained by $1 = \sum_w \phi_w$.

The deep mutational scanning data are incorporated into the ExpCM via the $F_{r,xy}$ terms.
The experiments measure the preference $\pi_{r,a}$ of every site $r$ for every amino-acid $a$.
The $F_{r,xy}$ terms are defined in terms of these experimentally measured amino-acid preferences as
\begin{equation}
\label{eq:Frxy}
F_{r,xy} = 
\begin{cases}
   1 & \mbox{if $\mathcal{A}\left(x\right) = \mathcal{A}\left(y\right)$} \\
   \omega \times \frac{\ln\left[\left(\pi_{r,\mathcal{A}\left(y\right)} / \pi_{r,\mathcal{A}\left(x\right)}\right)^{\beta}\right]}{1 - \left(\pi_{r,\mathcal{A}\left(x\right)} / \pi_{r,\mathcal{A}\left(y\right)}\right)^{\beta}} & \mbox{if $\mathcal{A}\left(x\right) \ne \mathcal{A}\left(y\right)$}
   \end{cases}
\end{equation}
where $\mathcal{A}\left(x\right)$ is the amino-acid encoded by codon $x$, $\beta$ is the stringency parameter, and $\omega$ is the relative rate of nonsynonymous to synonymous substitutions after accounting for the amino-acid preferences.
The ExpCM has six free parameters (three $\phi_w$ values, $\kappa$, $\beta$, and $\omega$).
The preferences $\pi_{r,a}$ are \emph{not} free parameters since they are determined by an experiment independent of the sequence alignment being analyzed.

The ExpCM stationary state frequency $p_{r,x}$ of codon $x$ at site $r$ is~\citep{bloom2017identification} 
\begin{equation}
\label{eq:p_rx}
p_{r,x} = \frac{\left(\pi_{r,\mathcal{A}\left(x\right)}\right)^{\beta} \phi_{x_0} \phi_{x_1} \phi_{x_2}}{\sum_z \left(\pi_{r,\mathcal{A}\left(z\right)}\right)^{\beta} \phi_{z_0} \phi_{z_1} \phi_{z_2}},
\end{equation}
\subsection*{Theoretical effect of model choice on branch length}
\subsection*{Effect of model choice on natural sequences}

\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/model_feature}}
\caption{\label{model_feature}
\textbf{Comparison of substitution model features.}
Site-specific amino-acid profiles and $\Gamma$-distributed rate variation are both substitution model features which have been shown or theorized to lengthen branches. 
The models YNGKP M0, YNGKP M5, ExpCM, and ExpCM+$\Gamma\omega$ represent all possible combinations of these two features. 
Blue indicates presence and white indicates absence of a feature. 
}
\end{figure}

\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/decay}}
\caption{\label{decay}
\textbf{ExpCMs infer longer branches than YNGKP models for sites with mutational constraint.}
The expected pairwise identity trajectories were calculated using \ref{eq:f} and models described in \ref{tab:decay_params}.
The trajectories of the YNKGP M0 (blue) and YNGKP M5 (green) do not vary from panel to panel because neither model is site-specific. 
The deviation in trajectory of the ExpCM (red) from the YNGKP M0 (blue) increases from left to right as the mutational constraint of the amino-acid profile increases (logoplots, above). 
The deviation in trajectory of the YNGKP model with a site-specific $\omega$ value inferred from the ExpCM (yellow, \ref{eq:w_r}) is also positively correlated with the constraint of the site-specific amino-acid profiles but the effect size is smaller.  
}
\end{figure}

\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/simulations}}
\caption{\label{simulations}
\textbf{Branch lengths simulated under an ExpCM are underestimated by YNGKP models and long branches are disproportionally affected.} 
Alignments were simulated under an ExpCM (\ref{tab:sim_params}) along an HA tree and the branches were re-optimized by a model from the ExpCM or YNGKP family. 
The randomized ExpCMs have amino-acid profiles shuffled among the sites 
These randomized models are still site-specific but the relationship between the site and the experimental data is broken. 
Grey points represent the length of one branch and the black points are the mean branch lengths over eight simulations. 
The grey, dashed line is the reference line $y=x$, depicting the behavior of a model which is an unbiased estimator of the simulated branch length. 
}
\end{figure}


\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/tree_doud}}
\caption{\label{fig:tree_doud}
\textbf{Trees optimized with an ExpCM defined by H1 preferences lengthen branches from the focal H1 sequence compared to YNGKP models.} 
The branch lengths of a base topology inferred using the GTR-CAT model were optimized by \textbf{(A)} an ExpCM defined by H1 preferences, \textbf{(B)} an ExpCM+$\Gamma\omega$ defined by H1 preferences, \textbf{(C)} YNKGP M0, and \textbf{(D)} YNGKP M5.
The branch lengths are normalized to the distance between A/South Carolina/1/1918 and A/Solomon Islands/3/2006 and colored to indicate the distance from the H1 focal sequence (black triangle).
}
\end{figure}

\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/tree_lee}}
\caption{\label{fig:tree_lee}
\textbf{Trees optimized with an ExpCM defined by H3 preferences lengthen branches from the focal H3 sequence compared to YNGKP models.} 
The branch lengths of a base topology inferred using the GTR-CAT model were optimized by \textbf{(A)} an ExpCM defined by H3 preferences, \textbf{(B)} an ExpCM+$\Gamma\omega$ defined by H3 preferences, \textbf{(C)} YNKGP M0, and \textbf{(D)} YNGKP M5.
The branch lengths are normalized to the distance between A/South Carolina/1/1918 and A/Solomon Islands/3/2006 and colored to indicate the distance from the H3 focal sequence (black circle).
}
\end{figure}

\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/tree_average}}
\caption{\label{fig:tree_average}
\textbf{Trees optimized with an ExpCM defined by the average of H1 and H3 preferences lengthen branches from both the focal H3 sequence and the focal H1 sequence compared to YNGKP models.} 
The branch lengths of a base topology inferred using the GTR-CAT model were optimized by \textbf{(A)} an ExpCM defined by the average preferences, \textbf{(B)} an ExpCM+$\Gamma\omega$ defined by the average preferences, \textbf{(C)} YNKGP M0, and \textbf{(D)} YNGKP M5.
The branch lengths are normalized to the distance between A/South Carolina/1/1918 and A/Solomon Islands/3/2006.
The black triangle indicates the H1 focal sequence and the black circle indicates the focal sequence.
}
\end{figure}

\section*{Conclusion}

\newpage
\section*{Materials and Methods}

\subsubsection*{Experimentally informed codon model (ExpCM)}

\subsubsection*{YNGKP M0}

\subsubsection*{ExpCM + $\Gamma\omega$ and YNGKP M5}


\subsubsection*{Spielman $\omega_{r}$ values inferred from the ExpCM} 
We inferred the average nonsynonymous fixation rate from the ExpCM following~\citet{spielman2015relationship} as 
\begin{equation}
\label{eq:w_r}
\omega_{r} = \frac{\sum_{x} \sum_{y \in N_x} {p_{r,x} \times P_{r,xy}}}{\sum_{x} \sum_{y \in N_x} {p_{r,x} \times Q_{xy}}}
\end{equation}
where $p_{r,x}$ is the stationary state of the ExpCM at site $r$ and codon $x$, $P_{r,xy}$ is the substitution rate from codon $x$ to codon $y$ at site $r$, $Q_{xy}$ is the mutation rate from codon $x$ to codon $y$, and $N_x$ is the set of codons that are nonsynonymous to codon $x$ and differ from codon $x$ by only one nucleotide. 

\subsubsection*{Expected pairwise amino-acid identity}
The expected pairwise amino-acid identity at a site $r$ over time $t$ for a given model is 
\begin{equation}
\label{eq:f}
\sum_a \sum_{x \in a} p_{r,x} \sum_{y \in a} [M_{r}\left(t\right)]_{xy}
\end{equation}
where $a$ is all amino acids, $p_{r,x}$ is the stationary state of the model at site $r$ and codon $x$, and $[M_{r}\left(t\right)]_{xy}$ is the transition rate from codon $x$ to codon $y$ at site $r$ given time $t$. 

\newpage
\section*{Supplemental Information}

\subsection*{Model Parameters for the simulations}

\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/prefs_doud}}
\caption{\label{fig:prefs_doud}
\textbf{H1 preferences measured by \cite{doud2016accurate} rescaled with the ExpCM stringency parameter optimized in \ref{fig:tree_doud}A  ($\beta = 1.21$)} 
}
\end{figure}

\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/prefs_lee}}
\caption{\label{fig:prefs_lee}
\textbf{H3 preferences measured by \textit{lee} rescaled with the ExpCM stringency parameter optimized in \ref{fig:tree_lee}A  ($\beta = 1.46$)} 
}
\end{figure}

\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/prefs_average}}
\caption{\label{fig:prefs_average}
\textbf{The average of the H1 preferences measured by \cite{doud2016accurate} and the H3 preferences measured by \textit{Lee} rescaled with the ExpCM stringency parameter optimized in \ref{fig:tree_average}A  ($\beta = 1.82$)}}
\end{figure}

\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/experiment_doud}}
\caption{\label{fig:experiment_doud}
\textbf{The ExpCM defined by H1 preferences lengthen longer branches on the HA tree.} 
\textbf{(A)} An HA alignment was subsampled to create three smaller alignments with varying degrees of divergence from the focal H1 sequence, referred to as "low", "intermediate", and "high". 
\textbf{(B)} A phylogenetic tree of the "high" alignment was constructed using the GTR-CAT model. 
The colors denote the alignment and the black circle denotes the focal H3 sequence. 
\textbf{(C)} The value of the ExpCM and ExpCM+$\Gamma\omega$ stringency parameter $\beta$ decreases as the divergence from the focal H1 sequence increases. 
\textbf{(D)} Comparisons of branch lengths optimized by the four substitution models for the varying degrees of divergence. 
Black points represent branches from the focal H3 sequence and grey points represent all other branches.  
The branch lengths are in average number of codon substitutions per site. 
}
\end{figure}

\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/experiment_lee}}
\caption{\label{fig:experiment_lee}
\textbf{The ExpCM defined by H1 preferences lengthen longer branches on the HA tree.} 
\textbf{(A)} An HA alignment was subsampled to create three smaller alignments with varying degrees of divergence from the focal H3 sequence, referred to as "low", "intermediate", and "high". 
\textbf{(B)} The phylogenetic tree of the "high" alignment. 
The colors denote the alignment and the black circle denotes the focal H3 sequence. 
\textbf{(C)} The value of the ExpCM and ExpCM+$\Gamma\omega$ stringency parameter $\beta$ decreases as the divergence from the focal H3 sequence increases. 
\textbf{(D)} Comparisons of branch lengths optimized by the four substitution models for the varying degrees of divergence. 
Black points represent branches from the focal H3 sequence and grey points represent all other branches.  
The branch lengths are in average number of codon substitutions per site. 
}
\end{figure}

\begin{figure}
\centerline{\includegraphics[width=0.85\textwidth]{figures/dms}}
\caption{\label{fig:dms}
\textbf{Schematic of deep mutational scanning.} 
\textbf{(A)} All single codon mutations are introduced into the wildtype HA gene. 
\textbf{(B)} Each virus in the mutant virus library contains one HA variant. 
\textbf{(C)} The mutant virus library is passaged in cell culture to select for functional variants. 
\textbf{(D)} Deep sequencing quantifies the frequency of each variant before and after selection. 
The preference for each amino acid at each site (as quantified by the deep sequencing) is represented as a logoplot.
}
\end{figure}

\begin{table}[t!]
\caption{\label{tab:simulation_params}
ExpCM parameters used to simulate sequences in Fig.~\ref{fig:simulation}.}
      \begin{tabular}{ccccc}
        \hline
          Parameter & Value\\ \hline
       	$\beta$ & $1.54$\\
	$\kappa$ & $3.60$\\
	$\omega$ & $0.20$\\
	$\phi_A$, $\phi_C$, $\phi_G$& $0.38$, $0.17$, $0.23$\\
      \end{tabular}
\end{table}

\begin{table}[t!]
\caption{\label{tab:decay_params}
Model parameters used in  Fig.~\ref{fig:decay}.}
      \begin{tabular}{ccccc}
        \hline
          Model & Parameters\\ \hline
          ExpCM & $\beta=2.0$, $\kappa=1.0$, $\omega=1.0$, $\phi_A=\phi_C=\phi_T=0.25$, $\pi_{r, A\left(X\right)}$:\cite{Doud2016accurate}\\
          YNGKP M0 & $\kappa=1.0$, $\omega=1.0$, $\phi_{rw}=0.25$\\
          YNGKP M5 & $\kappa=1.0$, $\alpha_\omega=0.36$, $\beta_\omega=1.9$, $\phi_{rw}=0.25$\\\
      \end{tabular}
\end{table}




\clearpage 
\bibliographystyle{mbe}
\bibliography{references.bib}



\end{document}