% Use only LaTeX2e, calling the article.cls class and 12-point type.

\documentclass[11pt]{article}

\usepackage[round,semicolon]{natbib}
\usepackage{etoolbox}
\AtBeginEnvironment{quote}{\singlespacing\tiny}
% Use times if you have the font installed; otherwise, comment out the
% following line.

% added by SKH
%\usepackage{lineno}
%\linenumbers

\usepackage{times}
\usepackage{amssymb}
\usepackage{amsmath}

\usepackage[export]{adjustbox}

\usepackage{graphicx}
\graphicspath{ {images/} }

% for adjustwidth
\usepackage{changepage}

% The following parameters seem to provide a reasonable page setup.

\topmargin 0.0cm
\oddsidemargin 0.2cm
\textwidth 16cm 
\textheight 21cm
\footskip 1.0cm

\usepackage{newfloat}
\usepackage{amsmath}
\usepackage[labelfont=bf]{caption}
\usepackage{nameref}
\usepackage{rotating}
\usepackage{color}
\usepackage{float}
\renewcommand{\figurename}{{}}
\renewcommand{\thefigure}{{Figure \arabic{figure}}}

\renewcommand{\tablename}{{}}
\renewcommand{\thetable}{{Table \arabic{table}}}

\newfloat{suppfile}{thp}{losuppfile}
\renewcommand{\thesuppfile}{Supplementary file \arabic{suppfile}}
\floatname{suppfile}{}

\newfloat{suppfig}{thp}{losuppfig}
\renewcommand{\thesuppfig}{Supplementary figure \arabic{suppfig}}
\floatname{suppfig}{}

%
\newfloat{supptable}{thp}{losupptable}
\renewcommand{\thesupptable}{Supplementary table \arabic{supptable}}
\floatname{supptable}{}
%

\renewcommand{\theequation}{Equation \arabic{equation}}

\newcommand{\mutDNA}{\textbf{mutDNA}}
\newcommand{\mutvirus}{\textbf{mutvirus}}
\newcommand{\DNA}{\textbf{DNA}}
\newcommand{\virus}{\textbf{virus}}

\newcommand\skhcomment[1]{{\color{cyan}[#1]}}
\newcommand\jdbcomment[1]{{\color{red}[#1]}}


\usepackage{hyperref}
\hypersetup{colorlinks,citecolor=blue,linkcolor=blue,urlcolor=blue}
\hypersetup{colorlinks,citecolor=blue,linkcolor=blue,urlcolor=blue}

\usepackage{seqsplit}

\usepackage{array}
\newcolumntype{P}[1]{>{\raggedright\arraybackslash}p{#1}}

\title{Experimentally informed site-specific substitution models deepen phylogenetic estimates of the divergence of viral lineages} 

\author
{Sarah K. Hilton$^{1,2}$  and Jesse D. Bloom$^{1,2}$\\
\\
\normalsize{$^1$Division of Basic Sciences and Computational Biology Program,}\\
\normalsize{Fred Hutchinson Cancer Research Center, Seattle, WA 98109, USA}\\
\normalsize{$^2$Department of Genome Sciences, University of Washington, Seattle, WA}\\
\normalsize{E-mail:  jbloom@fredhutch.org.}\\
}


% Include the date command, but leave its argument blank.

\date{}

\usepackage{setspace}
\onehalfspacing


\begin{document} 

% Make the title.

\maketitle 


\begin{abstract}
\noindent  
Molecular phylogenetics is often used to estimate the time since the divergence of modern gene sequences.
Such phylogenetic techniques often estimate substantially shallower divergence times than other methods.
For instance, in the case of viruses there is independent evidence that molecular phylogenetics can underestimate deep divergence times.  
This discrepancy is thought to be caused in part by inadequate models of purifying selection leading to branch-length underestimation.
Here we show that substitution models informed by experimental measurements of the purifying selection due to site-specific amino-acid preferences lengthen the deep branches on phylogenies of influenza virus hemagglutinin.
This deepening of branch lengths is due to better modeling of the stationary state of the substitution models, and is independent of the branch-length-extension that results from modeling site-to-site variation in substitution rate.
The deepening of branch lengths from experimentally informed site-specific substitution models is similar to that achieved by other approaches that allow the stationary state to vary across sites.
However, the improvements from these site-specific models are limited by the inherent tension between the enhanced accuracy of accounting for site-specific amino-acid preferences and the fact that these preferences shift over long evolutionary times.
Overall, our work underscores the importance of modeling how site-specific purifying selection affects the stationary state when estimating deep divergence times. 
\end{abstract}

\clearpage

\section*{Introduction} 
\skhcomment{from JDB: what is the "age" of a virus? Maybe "divergence time of viral lineages"}
skhcomment{from JDB: what is the less than a million actually? "Old" is not the right phrase.}
Estimating the divergence time of viral lineages of a virus is essential to understanding its evolutionary history, including its emergence, spread, and past zoonoses. 
This estimation is commonly done using the concept a ``molecular clock" to transform the branch lengths of the viral phylogenetic tree into age in years. 
However, this molecular dating technique often underestimates the age of many viruses, including measles, foamy virus, and ebola \skhcomment{(citations)}, compared to other methods which are independent of the viral phylogeny. 
For example, SIV (the original source of HIV) is estimated to be less than a million years old based on the viral phylogeny \citep{sharp2000origins,wertheim2009dating,worobey2010island} but estimated to be several million years old based on the host tree or endogenous retroviral elements  \citep{compton2013host} \skhcomment{(other citations)}. 
Overall, there is a systematic and substantially large underestimation of of branch length on viral phylogenies. 
\skhcomment{long branches}

Branch length underestimation is due, in part, to strong purifying selection masking the evolutionary signal in the observed sequences. 
Purifying selection can lead to mutational saturation, where multiple unobserved, substitutions occur at a single site along a long branch and erase the divergence signal \citep{holmes2003molecular}.
Furthermore, proteins do not have equal preference for all amino acids at all sites, this evident by a simple visual inspection of a multiple sequence alignment. 
How many and which amino acids tolerated at each site of the protein generate a site-specific expected rate of change. 
Failing to account for these site-specific constraints will lead to branch length underestimation. 
\skhcomment{you will have mutational saturation no matter what - this is a separate, addressable issue?}
\skhcomment{talk about the high mutation rate in viruses?}

Substitution models that incorporate site-to-site rate variation have been developed to decrease the bias in long branch estimation. 
The most common strategy is to allow a single rate-controlling parameter to vary according to some statistical distribution, such as a $\Gamma$-distributed $\omega$ (~dN/dS) \citep{yang2000codon}. 
This flexibility in the value of $\omega$ accounts for the site-to-site rate variation by allow some sites to have a higher dN/dS value than others. 
While this modification is simple and only requires the addition of one extra parameter, it does not describe site-specificity in its stationary state. 
That is, at evolutionary equilibrium, this model still assumes that each site in the protein evolves identically.  

An alternative approach is to model the site-specific amino-acid frequencies explicitly, such as those models in the mutation-selection family \citep{halpern1998evolutionary}. 
In these models, each amino-acid at each site in the protein is described by its own parameter and these differences are reflected in the stationary state of the model. 
The rate of change at a given site is controlled by these amino acid profiles and can now vary from site to site, as expected based on observations in nature. 
Importantly, these rate variations are not constrained to an arbitrary statistical distribution but by parameters with a direct biological interpretation. 

Mutation-selection models are presumably more biologically relevant but pose more practical challenges than the $\Gamma\omega$ models. 
These models are highly parametrized with 19 free parameters (the 20 amino acid preferences are constrained to sum to one) per site leading to thousands of parameters for the length of a normal protein. 
One way to avoid overfitting is to implement the model as a mixture model in either a bayesian \citep{lartillot2004bayesian} or maximum likelihood framework \citep{si2008empirical}. 

Alternatively, you can reduce the parameter space by defining the amino-acid frequencies \textit{a priori}. 
We have shown previously that we can define an Experimentally Informed Codon Model (ExpCM) \citep{bloom2014experimentally,bloom2014informed} from the mutation-selection family using measurements from deep mutational scanning \citep{fowler2014deep}, a high-throughput functional assay. 
ExpCM are therefore defined by amino-acid preferences measured in a \textit{single} genetic background and do not reflect any epistatic changes which may have occurred over the virus's evolutionary history. 
But they contain no more parameters than the traditional codon models while maintaining a site-specific stationary state. 
We hypothesize that the ExpCM will estimate longer branches than the traditional models due to the protein-specific description of purifying selection. 
\skhcomment{CAT model has been shown to work well (better) on saturated data.}

In order to test this hypothesis, we compared the branch lengths of a influenza virus HA phylogenetic trees optimized by different substitution models. 
We found that the ExpCM did extend the length of branches from the focal sequence on the tree \skhcomment{define focal} and that this extension was seen even in the context of $\Gamma$-distributed rate variation. 
Furthermore, we found this extension occurred even in the presence of $\Gamma$-distributed $\omega$, indicating that they are both important for modeling purifying selection. 
This supports the conclusion that modeling purifying selection, especially in a model with a non-uniform stationary state, is important to estimating the branch lengths on phylogenetic trees. 

\section*{Results and Discussion}

\subsection*{Different ways hat substitution models account for purifying selection}

\jdbcomment{Some other comments on this section, which I think in general is pretty good: $\omega$ shows up in the figure, but is never mentioned in the text. I feel like GY94 models need to be explained at least in terms of what they and $\Gamma$ distribution stand for. Just introducing nomenclature. I feel like the last paragraph should tie back to panel C of the figure.}

\begin{figure}[H]
\centerline{\includegraphics[width=0.90\textwidth]{figures/model_feature.pdf}}
\caption{\label{fig:model_feature}
\textbf{Different ways of modeling site-to-site variation in purifying selection.}
(A) Substitution model stationary states can either be identical at every site in the protein or allow to vary from site to site. 
(B) The relative rate of non-synonymous change, $\omega$, can be defined as one, gene-wide average or allowed to vary following some statistical distribution.
In order to achieve computational tractability, the distribution is discretized into $K$ bins and $\omega$ is set to the mean of each bin. 
A $\Gamma$ distribution with $K=4$ bins is shown here. 
(C) Substitution models can both, one, or neither of these features. 
We used models from the GY94 and the ExpCM families. 
}
\end{figure}

Proteins evolve under both purifying selection to maintain their structure and function. 
This purifying selection is not homogenous across sites in a protein.
It is also not homogenous across the different amino acids at a given site.
For instance, some protein sites strongly prefer hydrophobic amino acids, others may be constrained to just one or a few amino acids, and yet others may tolerate many amino acids.
In general, these constraints are highly idiosyncratic among sites, and so pose a challenge for phylogenetic substitution models.

The most common strategy to model rate variation is to allow the rate of non-synonymous change to vary among sites following some statistical distribution \citep[\ref{fig:model_feature}A;][]{yang1994maximum, yang2000codon}. 
Under such models, some sites have a higher rate of non-synonymous change than others. 
However, this rate is agnostic to the amino-acid identities themselves, and so all non-synonymous changes at at given site are treated equally. 

In contrast, so-called ``mutation-selection'' models~\citep{halpern1998evolutionary} account for purifying selection by explicitly defining a different set of amino-acid preferences at each site in the protein. 
This more mechanistic formulation, without an arbitrary statistical distribution, results in a site-specific stationary state (\ref{fig:model_feature}B). 
These models better capture the site-to-site differences in amino-acid composition that is an obvious features of real proteins, and indeed they generally better describe actual evolution than models with only rate variation~\citep{lartillot2004bayesian, lartillot2007suppression, le2008phylogenetic, rodrigue2010mutation,si2008empirical,rodrigue2014site,bloom2017identification,hilton2017phydms,bloom2014experimentally}.
The specificity of mutation-selection models comes at a cost in the form of an increased number of parameters. 
While codon substitution models with uniform stationary states typically have $<$20 parameters, mutation-selection models must specify 19 parameters for \emph{each} site (the stationary state is for 20 amino acids whose frequencies are constrained to sum to one).
This corresponds to $19\times L$ parameters for a protein of length $L$, which leads to $\sim 10^4$ parameters for a typical size protein.
As with all parameter-rich models, it is important to obtain values for these parameters using some method that avoids overfitting.
Here we will primarily use experimentally informed codon models (ExpCM) which define values for these parameters \textit{a priori} from deep mutational scanning experiments~\citep{araya2011,fowler2010high} so they do not need to be fit from phylogenetic data~\citep{bloom2014experimentally, hilton2017phydms, bloom2017identification}.
Therefore, for an ExpCM the number of remaining free parameters that are fit from the phylogenetic data are similar to a non-site-specific substitution model.
Alternative strategies of obtaining parameters for site-specific stationary states via Bayesian\skhcomment{cite} or maximum-likelihood estimation\skhcomment{cite} are discussed in the last section of the Results.

These two strategies to account for site-to-site variation in purifying selection are not mutually exclusive. 
\jdbcomment{I'm not sure if I love the rest of this paragraph. I feel like GY94 needs to be introduce more directly---perhaps in the paragraph beginning the ``The most common strategy...'' above. Also, it's a bit unclear what ``most appropriate'' means.}
While differences in stationary state manifest as differences in non-synonymous rates across sites, these models are most appropriate for modeling purifying selection \citep{spielman2015relationship}.
Proteins which are under strong purifying and diversifying selection may be better modeled by a combination of the two. 
For this study, we used four models from the GY94 and ExpCM families, which together represent all possible combinations of these two strategies (\ref{fig:model_feature}C). 

\subsection*{Effect of stationary state and rate variation on branch length estimation}

\begin{figure}[H]
\centerline{\includegraphics[width=0.90\textwidth]{figures/decay.pdf}}
\caption{\label{fig:decay}
\textbf{Effect of stationary state and rate variation on long branch estimation.}
The expected pairwise amino-acid identity for five sites in influenza hemagglutinin (HA) for four different substitution models. 
The logoplots show the HA amino-acid preferences from a deep mutational scan performed by \cite{doud2016accurate}. 
The implied dN/dS value was calculated from the ExpCM following \cite{spielman2015relationship}.
\skhcomment{calculated using equation. Maybe change long branch length to expected asymptotic sequence divergence.}
}
\end{figure}

Given a single branch, a substitution model transforms sequence divergence into branch length. 
Under the molecular clock assumption, this branch length is proportional to time. 
However, every substitution model has a stationary state, defining the expected sequence composition after a long evolutionary time. 
This stationary state can be used to calculate the expected amino-acid divergence a very long amount of time has elapsed. 
(Rate only affects how fast asymptotic divergence approached, whereas). 
In \ref{fig:decay}, the stationary state is represented by the ``long" tails where the expected sequence divergence remains constant as time increases. 

This is a fundamental feature. 
The stationary state is principle eigenvector of substitution matrix. 
Rate variation only multiples by a number, and stationary state of a stochastic matrix is invariant to multiplication. 

Along a single branch, a substitution model transforms sequence divergence into branch length. 
Under the molecular clock assumption, this branch length is proportional to time.  
However, every substitution model has a stationary state, defining the expected sequence divergence given a very long time, at which this relationship between sequence divergence and time is lost. 
In \ref{fig:decay}, the stationary state is represented by the long ``tails" where the expected sequence divergence remains constant as time increases. 

The specific parametrization scheme of a model will control features of its stationary state. 
Comparing GY94 to GY94+$\Gamma\omega$ will change how long it takes for the model to reach the stationary state but it does not change the expected sequence divergence at the stationary state. 
Furthermore, the GY94 and GY94+$\Gamma\omega$ stationary state is exactly the same at each site in the protein. 
Site-specific stationary states modulate both the amount of time to reach and the overall divergence level at the stationary state. 
These differences arise from whether or not the site is modeled as constrained or mutationally tolerant.  
The stationary state is important to branch length estimation because differences in expected divergence will lead to differences in branch length estimation and branch lengths cannot be estimated with any accuracy when two sequences have reached the divergence level of the stationary state. 

The difference in branch length estimation between uniform and site-specific stationary state models will be most dramatic at constrained sites. 
Most uniform stationary state models define roughly equal frequency of each amino acid, in essence they assume that each site is mutationally tolerant. 
If the sequence divergence at constrained site is quite small, a site-uniform model will expect a small amount of time to have passed, a short branch, and a site-specific model will expect that a large amount of time has passed, long branch. 
At a mutationally tolerant site, both models will estimate roughly the same branch length for any given divergence. 

This difference in branch length estimation cannot be recapitulated by more complex modeling of the $\omega$ parameter. 
We transformed the GY94 model into a site-specific model by inferring a unique $\omega$ value from the ExpCM for each site in the protein, following the work of \cite{spielman2015relationship}. 
This model takes longer to reach stationary state than the GY94 model but, ultimately, has the same divergence level at stationary state. 
As this model shows, no matter how ``well" a model accounts for site-to-site rate variation, it will underestimate long branches with a uniform stationary state model.  

\subsection*{Failure to account for site-specificity leads to branch length underestimation.}

\begin{figure}[H]
\centerline{\includegraphics[width=0.85\textwidth]{figures/simulations}}
\caption{\label{simulations}
\textbf{Model performance under simulated, site-specific data.} 
Alignments were simulated under an ExpCM along an HA tree and the branches were re-optimized by a model from the ExpCM or YNGKP family. 
The averaged ExpCMs amino-acid frequencies defined by the average preference of that amino acid across all sites in the protein. 
While these models extract information from the deep mutational scanning experiment, they are not site-specific. 
Grey points represent the length of one branch and the black points are the mean branch lengths over ten simulations. 
The grey, dashed line is the reference line $y=x$, depicting the behavior of a model which is an unbiased estimator of the simulated branch length. 
}
\end{figure}

To test the effect of substitution models on branch length estimation given sequences with site-specific amino-acid frequencies we simulated sequences under an ExpCM and re-inferred the branch lengths using the different substitution models described in \ref{fig:model_feature}C.
As expected, the ExpCM and ExpCM+$\Gamma\omega$ estimated the branch lengths of the simulated sequences accurately. 
The variance in the estimation of an given branch increases between the simulations increased as branch length increased but this error is not biased towards over- or underestimation.
The models with a uniform stationary state consistently underestimate the long branches. 
This underestimation is seen with both the GY94 and the GY94+$\Gamma\omega$, indicating that accounting for rate variation via the rate parameter cannot prevent branch length underestimation at these long branches. 
The ExpCM can also be transformed into a uniform stationary state model by setting the amino-acid at every site to the average value for that amino acid across all sites. 
In contrast to the GY94 models, this average control does not have preference for amino acid and does extract some information from the experiments, but like the GY94 models the stationary state is uniform across sites. 
These models performed similarly to the GY94 and GY94+$\Gamma\omega$ and underestimated long branches. 

These simulations show that when the sequences have site-specific amino-acid frequencies, models with uniform stationary states cannot accurately estimate branch lengths and, will in fact, underestimate long branches. 	

\subsection*{empirical Data}

\begin{figure}[H]
\centerline{\includegraphics[width=0.8\textwidth]{figures/empirical_trees.pdf}}
\caption{\label{fig:empirical_trees}
\textbf{Trees optimized with an ExpCM defined by H1 preferences lengthen branches from the focal H1 sequence compared to GY94 models.} ).
}
\end{figure}

\begin{table}[t!]
\caption{\label{tab:empirical_data}
Model comparison for Fig.~\ref{fig:empirical_trees}.\skhcomment{GY94+$\Gamma\omega$ has one $\omega$ of 0 because of rounding.}} 
     \begin{tabular}{cccccccccc}
        \hline
         Model & {\shortstack{Stationary\\ State}} & $\Gamma\omega$ & $\Delta$AIC & {\shortstack{Log\\ Likelihood}} & {\shortstack{$\omega$\\ (implied dN/dS)}} & {\shortstack{stringency\\ parameter}}\\ \hline
       	ExpCM + $\Gamma\omega$ (H1+H3 avg) & yes & yes & 0 & -487510 & 0.19,  0.50,  0.90,  1.86 &  1.70, \\
	ExpCM (H1+H3 avg) & yes & no &  950 & -492270 & 0.15 & 1.78\\
	ExpCM + $\Gamma\omega$ (H1) & yes & yes & 13060 & -494040  & 0.13 ,  0.44,  0.91,  2.16 & 1.12\\
	ExpCM + $\Gamma\omega$ (H3) & yes & yes & 17370 & -49620 & 0.09,  0.33,  0.72,  1.77 & 1.28\\
	ExpCM (H1) & yes & no & 2556 & -50030 &  0.13 & 1.22\\
	ExpCM (H3) & yes & no &  3197 & -50350 & 0.12 & 1.45\\
	GY94 + $\Gamma\omega$ & no & yes & 4719 & -51106 & 0.00,  0.03,  0.08,  0.26 & N/A \\
	GY94 & no & no & 7625 & -52560  & 0.07 & N/A\\
      \end{tabular}
\end{table}

\skhcomment{Section Outline:}

\begin{itemize}
\item We tested the effect substitution model choice on branch length estimation using influenza HA sequences. 
\begin{itemize}
\item These sequences presumably evolve with site-specific frequencies but unlike the simulations we do not \textit{know} the model
\item These sequences are from fairly diverged proteins. Some sequences share only 40\% identity on the amino acid level 
\item The HA tree has closely related subtypes with long branches between subtypes. This is good because we are looking for the effect on long branches. 
\end{itemize}
\item Both $\Gamma\omega$ and non-uniform stationary states result in an extension of branch lengths. 
\begin{itemize}
\item The addition of $\Gamma\omega$ extends branches independent of preferences. 
\item The addition of preferences extends branches \textit{from the focal sequence} independent of $\Gamma\omega$. 
\end{itemize}
\item The extension of branch length from the focal sequence shows that the ExpCM does not have equal "relevance" across the whole tree. 
\begin{itemize}
\item This result is not entirely surprising. DMS measure the effects of single amino-acid changes in the context of a single genetic background. 
\item It is expected that there would be epistatic changes across such a highly diverged phylogenetic tree. ``Shifting preferences".
\item "Sampling" from both preference sets via preference averaging extends the branch length from both focal sequences.
\end{itemize}
\end{itemize}

\subsection*{Competing effects of shifting preferences and long branches.}

\begin{figure}[H]
\centerline{\includegraphics[width=0.85\textwidth]{figures/doud_compete_2}}
\caption{\label{fig:doud_compete}
\textbf{The ExpCM defined by H1 preferences lengthen longer branches on the HA tree.} 
\textbf{(A)} An HA alignment was subsampled to create three smaller alignments with varying degrees of divergence from the focal H3 sequence, referred to as "low", "intermediate", and "high". 
\textbf{(B)} The phylogenetic tree of the "high" alignment. 
The colors denote the alignment and the black circle denotes the focal H3 sequence. 
\textbf{(C)} The value of the ExpCM and ExpCM+$\Gamma\omega$ stringency parameter $\beta$ decreases as the divergence from the focal H3 sequence increases. 
\textbf{(D)} Comparisons of branch lengths optimized by the four substitution models for the varying degrees of divergence. 
Black points represent branches from the focal H3 sequence and grey points represent all other branches.  
The branch lengths are in average number of codon substitutions per site. 
}
\end{figure}

\skhcomment{Section Outline:}

\begin{itemize}
\item We investigated the competing effects of ``shifting preferences" and long branches by comparing ExpCMs with trees of increasing overall divergence from the focal sequence. 
\item We use the fit stringency parameter $\beta$ as a measure of relevance. 
\begin{itemize}
\item The stringency parameter rescales the preferences. A stringency parameter with a value greater than one indicates that selection in nature prefers the same amino acids as selection in lab but with greater strength. 
\item There is an inverse relationship between the stringency parameter and overall sequence divergence. As the overall divergence of the tree increases, the stringency parameter decreases. 
\end{itemize}
\item \skhcomment{hypothetical data} There is also a positive relationship between the stringency parameter and branch lengths 
\begin{itemize}
\item We optimized the same tree with ExpCMs with different stringency parameters. 
\item The ExpCMs with higher stringency values estimate longer branches than the ExpCMs with lower stringency values. 
\end{itemize}
\end{itemize}

\subsection*{\texttt{phylobayes}}

\skhcomment{Section Outline}

\begin{itemize}
\item \texttt{phylobayes} implements the mutation-selection model in a bayesian framework as a way to avoid overfitting. 
\item \texttt{phylobayes} and ExpCM (H1+H3 average) estimate branches of roughly equal length 
\begin{itemize}
\item the ExpCM estimate longer branches from the focal sequences 
\item \texttt{phylobayes} estimates longer branches from the non-focal sequences 
\end{itemize}
\end{itemize}

\begin{figure}[H]
\centerline{\includegraphics[width=0.5\textwidth]{figures/phylobayes.pdf}}
\caption{\label{fig:phylobayes}
\textbf{Comparison of ExpCM and phylobayes}
\skhcomment{y=x line too faint?}
}
\end{figure}

\section*{Conclusion}

\begin{enumerate}
  \item We don't allow any of the models to vary by lineage. 
\end{enumerate}

\newpage
\section*{Materials and Methods}

\subsection*{Substitution models}
\subsubsection*{GY94 models}
\subsubsection*{ExpCMs}
We recap the \textbf{Exp}erimentally Informed \textbf{C}odon \textbf{M}odel (ExpCM) \citep{bloom2014experimentally,bloom2014informed,bloom2017identification,hilton2017phydms} to introduce nomenclature. 

In an ExpCM, rate of substitution $P_{r,xy}$ of site $r$ from codon $x$ to $y$ is written in mutation-selection form~\citep{halpern1998evolutionary,mccandlish2014modeling,spielman2015relationship} as
\begin{equation}
P_{r,xy} = Q_{xy} \times F_{r,xy}
\end{equation}
where $Q_{xy}$ is proportional to the rate of mutation from $x$ to $y$, and $F_{r,xy}$ is proportional to the probability that this mutation fixes.
The rate of mutation $Q_{xy}$ is assumed to be uniform across sites, and takes an HKY85-like~\citep{hasegawa1985dating} form:
\begin{equation}
Q_{xy} = 
\begin{cases}
\phi_w & \mbox{if $x$ and $y$ differ by a transversion to nucleotide $w$} \\
\kappa \phi_w & \mbox{if $x$ and $y$ differ by a transition to nucleotide $w$} \\
0 & \mbox{if $x$ and $y$ differ by $>1$ nucleotide.}
\end{cases}
\end{equation}
The $\kappa$ parameter represents the transition-transversion ratio, and the $\phi_w$ values give the expected frequency of nucleotide $w$ in the absence of selection on amino-acid substitutions, and are constrained by $1 = \sum_w \phi_w$.

The deep mutational scanning data are incorporated into the ExpCM via the $F_{r,xy}$ terms.
The experiments measure the preference $\pi_{r,a}$ of every site $r$ for every amino-acid $a$.
The $F_{r,xy}$ terms are defined in terms of these experimentally measured amino-acid preferences as
\begin{equation}
\label{eq:Frxy}
F_{r,xy} = 
\begin{cases}
   1 & \mbox{if $\mathcal{A}\left(x\right) = \mathcal{A}\left(y\right)$} \\
   \omega \times \frac{\ln\left[\left(\pi_{r,\mathcal{A}\left(y\right)} / \pi_{r,\mathcal{A}\left(x\right)}\right)^{\beta}\right]}{1 - \left(\pi_{r,\mathcal{A}\left(x\right)} / \pi_{r,\mathcal{A}\left(y\right)}\right)^{\beta}} & \mbox{if $\mathcal{A}\left(x\right) \ne \mathcal{A}\left(y\right)$}
   \end{cases}
\end{equation}
where $\mathcal{A}\left(x\right)$ is the amino-acid encoded by codon $x$, $\beta$ is the stringency parameter, and $\omega$ is the relative rate of nonsynonymous to synonymous substitutions after accounting for the amino-acid preferences.
The ExpCM has six free parameters (three $\phi_w$ values, $\kappa$, $\beta$, and $\omega$).
The preferences $\pi_{r,a}$ are \emph{not} free parameters since they are determined by an experiment independent of the sequence alignment being analyzed.

The ExpCM stationary state frequency $p_{r,x}$ of codon $x$ at site $r$ is~\citep{bloom2017identification} 
\begin{equation}
\label{eq:p_rx}
p_{r,x} = \frac{\left(\pi_{r,\mathcal{A}\left(x\right)}\right)^{\beta} \phi_{x_0} \phi_{x_1} \phi_{x_2}}{\sum_z \left(\pi_{r,\mathcal{A}\left(z\right)}\right)^{\beta} \phi_{z_0} \phi_{z_1} \phi_{z_2}},
\end{equation}
\subsection*{Theoretical effect of model choice on branch length}
\subsection*{Effect of model choice on natural sequences}

\subsubsection*{ExpCM + $\Gamma\omega$ and YNGKP M5}


\subsubsection*{Spielman $\omega_{r}$ values inferred from the ExpCM} 
We inferred the average nonsynonymous fixation rate from the ExpCM following~\citet{spielman2015relationship} as 
\begin{equation}
\label{eq:w_r}
\omega_{r} = \frac{\sum_{x} \sum_{y \in N_x} {p_{r,x} \times P_{r,xy}}}{\sum_{x} \sum_{y \in N_x} {p_{r,x} \times Q_{xy}}}
\end{equation}
where $p_{r,x}$ is the stationary state of the ExpCM at site $r$ and codon $x$, $P_{r,xy}$ is the substitution rate from codon $x$ to codon $y$ at site $r$, $Q_{xy}$ is the mutation rate from codon $x$ to codon $y$, and $N_x$ is the set of codons that are nonsynonymous to codon $x$ and differ from codon $x$ by only one nucleotide. 

\subsubsection*{Expected pairwise amino-acid identity}
\textit{Do I need to talk about the branchScale scaling I used?}
The expected pairwise amino-acid identity at a site $r$ over time $t$ for a given model is 
\begin{equation}
\label{eq:f}
\sum_a \sum_{x \in a} p_{r,x} \sum_{y \in a} [M_{r}\left(t\right)]_{xy}
\end{equation}
where $a$ is all amino acids, $p_{r,x}$ is the stationary state of the model at site $r$ and codon $x$, and $[M_{r}\left(t\right)]_{xy}$ is the transition rate from codon $x$ to codon $y$ at site $r$ given time $t$. 

\newpage
\section*{Supplemental Information}

\subsection*{Model Parameters for the simulations}

\begin{suppfig}[H]
\centerline{\includegraphics[width=\textwidth]{figures/prefs_doud}}
\caption{\label{suppfig:prefs_doud}
\textbf{H1 preferences measured by \cite{doud2016accurate} rescaled with the ExpCM stringency parameter optimized in \ref{fig:tree_doud}A  ($\beta = 1.19$)} 
\skhcomment{I need to change the $\beta$ value when the new \texttt{phydms} results finish running.}
}
\end{suppfig}

\begin{suppfig}[H]
\centerline{\includegraphics[width=\textwidth]{figures/prefs_lee}}
\caption{\label{suppfig:prefs_lee}
\textbf{H3 preferences measured by \textit{lee} rescaled with the ExpCM stringency parameter optimized in \ref{fig:tree_lee}A  ($\beta = 1.46$)}
\skhcomment{I need to change the $\beta$ value when the new \texttt{phydms} results finish running.} 
}
\end{suppfig}

\begin{suppfig}[H]
\centerline{\includegraphics[width=\textwidth]{figures/prefs_average}}
\caption{\label{fig:prefs_average}
\textbf{The average of the H1 preferences measured by \cite{doud2016accurate} and the H3 preferences measured by \textit{Lee} rescaled with the ExpCM stringency parameter optimized in \ref{fig:tree_average}A  ($\beta = 1.77$)}}
\skhcomment{I need to change the $\beta$ value when the new \texttt{phydms} results finish running.}
\end{suppfig}
 


\begin{table}[t!]
\caption{\label{tab:simulation_params}
ExpCM parameters used to simulate sequences in Fig.~\ref{fig:simulation}.}
      \begin{tabular}{ccccc}
        \hline
          Parameter & Value\\ \hline
       	$\beta$ & $1.54$\\
	$\kappa$ & $3.60$\\
	$\omega$ & $0.20$\\
	$\phi_A$, $\phi_C$, $\phi_G$& $0.38$, $0.17$, $0.23$\\
      \end{tabular}
\end{table}

\begin{table}[t!]
\caption{\label{tab:wsn_low_params}
Model parameters used in  Fig.~\ref{fig:decay}.}
      \begin{tabular}{ccccc}
        \hline
          Model & Parameters\\ \hline
          ExpCM & $\beta=1.54196$\\
           & $\kappa=3.47184$\\
           & $\omega=0.219225$\\ 
          YNGKP M0 & $\kappa=2.9984$\\
          & $\omega=0.09076$\\
          YNGKP M5 & $\kappa=2.9984$\\
          & $\omega=0.09076$\\
      \end{tabular}
\end{table}

\begin{suppfig}[H]
\centerline{\includegraphics[width=0.85\textwidth]{figures/lee_compete}}
\caption{\label{fig:lee_compete}
\textbf{The ExpCM defined by H1 preferences lengthen longer branches on the HA tree.} 
\textbf{(A)} An HA alignment was subsampled to create three smaller alignments with varying degrees of divergence from the focal H3 sequence, referred to as "low", "intermediate", and "high". 
\textbf{(B)} The phylogenetic tree of the "high" alignment. 
The colors denote the alignment and the black circle denotes the focal H3 sequence. 
\textbf{(C)} The value of the ExpCM and ExpCM+$\Gamma\omega$ stringency parameter $\beta$ decreases as the divergence from the focal H3 sequence increases. 
\textbf{(D)} Comparisons of branch lengths optimized by the four substitution models for the varying degrees of divergence. 
Black points represent branches from the focal H3 sequence and grey points represent all other branches.  
The branch lengths are in average number of codon substitutions per site. 
}
\end{suppfig}


\clearpage 
\bibliographystyle{mbe}
\bibliography{references.bib}



\end{document}