% Use only LaTeX2e, calling the article.cls class and 12-point type.

\documentclass[11pt]{article}

\usepackage[round,semicolon]{natbib}
\usepackage{etoolbox}
\AtBeginEnvironment{quote}{\singlespacing\tiny}
% Use times if you have the font installed; otherwise, comment out the
% following line.

% added by SKH
\usepackage{lineno}
%\linenumbers

\usepackage{times}
\usepackage{amssymb}
\usepackage{amsmath}

\usepackage[export]{adjustbox}

\usepackage{graphicx}
\graphicspath{ {images/} }

% for adjustwidth
\usepackage{changepage}

% The following parameters seem to provide a reasonable page setup.

\topmargin 0.0cm
\oddsidemargin 1cm
\textwidth 15cm 
\textheight 21cm
\footskip 1.0cm

\usepackage{newfloat}
\usepackage{amsmath}
\usepackage[labelfont=bf]{caption}
\usepackage{nameref}
\usepackage{rotating}
\usepackage{color}
\usepackage{float}

% allow bigger floats per here: https://tex.stackexchange.com/a/11382
\renewcommand{\topfraction}{.95}
\renewcommand{\bottomfraction}{.95}
\renewcommand{\textfraction}{.05}
\renewcommand{\floatpagefraction}{.95}
\renewcommand{\dbltopfraction}{.66}
\renewcommand{\dblfloatpagefraction}{.66}
\setcounter{topnumber}{9}
\setcounter{bottomnumber}{9}
\setcounter{totalnumber}{20}
\setcounter{dbltopnumber}{9}

\renewcommand{\figurename}{{}}
\renewcommand{\thefigure}{{Figure~\arabic{figure}}}

\renewcommand{\tablename}{{}}
\renewcommand{\thetable}{{Table~\arabic{table}}}

\newfloat{suppfile}{thp}{losuppfile}
\renewcommand{\thesuppfile}{Supplementary~file~\arabic{suppfile}}
\floatname{suppfile}{}

\newfloat{suppfig}{thp}{losuppfig}
\renewcommand{\thesuppfig}{Supplementary~figure~\arabic{suppfig}}
\floatname{suppfig}{}

%
\newfloat{supptable}{thp}{losupptable}
\renewcommand{\thesupptable}{Supplementary~table~\arabic{supptable}}
\floatname{supptable}{}
%

\renewcommand{\theequation}{Equation~\arabic{equation}}

\newcommand\skhcomment[1]{{\color{cyan}[#1]}}
\newcommand\jdbcomment[1]{{\color{red}[#1]}}


\usepackage{hyperref}
\hypersetup{colorlinks,citecolor=blue,linkcolor=blue,urlcolor=blue}
\hypersetup{colorlinks,citecolor=blue,linkcolor=blue,urlcolor=blue}

\usepackage{seqsplit}

\usepackage{array}
\newcolumntype{P}[1]{>{\raggedright\arraybackslash}p{#1}}

\title{Modeling site-specific amino-acid preferences deepens phylogenetic estimates of viral sequence divergence} 

\author
{Sarah K. Hilton$^{1,2}$  and Jesse D. Bloom$^{1,2*}$\\
\\
\footnotesize{$^1$Basic Sciences and Computational Biology Program, Fred Hutchinson Cancer Research Center}\\
\footnotesize{$^2$Department of Genome Sciences, University of Washington}\\
\footnotesize{Seattle, WA, USA}\\
\footnotesize{$^*$E-mail:  jbloom@fredhutch.org.}\\
}


% Include the date command, but leave its argument blank.

\date{}

\usepackage{setspace}
\onehalfspacing


\begin{document} 

% Make the title.

\maketitle 


\begin{abstract}
\noindent  
Molecular phylogenetics is often used to estimate the time since the divergence of modern gene sequences.
For highly diverged sequences, such phylogenetic techniques sometimes estimate surprisingly recent divergence times. 
In the case of viruses, independent evidence indicates that the estimates of deep divergence times from molecular phylogenetics are sometimes too recent.
This discrepancy is caused in part by inadequate models of purifying selection leading to branch-length underestimation.
Here we examine the effect on branch-length estimation of using models that incorporate experimental measurements of purifying selection.
We find that models informed by experimentally measured site-specific amino-acid preferences estimate longer deep branches on phylogenies of influenza virus hemagglutinin.
This lengthening of branches is due to more realistic stationary states of the models, and is mostly independent of the branch-length-extension from modeling site-to-site variation in amino-acid substitution rate.
The branch-length extension from experimentally informed site-specific models is similar to that achieved by other approaches that allow the stationary state to vary across sites.
However, the improvements from all of these site-specific but time-homogeneous and site-independent models are limited by the fact that a protein's amino-acid preferences gradually shift as it evolves.
Overall, our work underscores the importance of modeling site-specific amino-acid preferences when estimating deep divergence times---but also shows the inherent limitations of approaches that fail to account for how these preferences shift over time.
\end{abstract}

\clearpage

\section*{Introduction} 
Molecular phylogenetics is commonly used to estimate the historical timing of evolutionary events \citep{yang2012molecular}.
This is done by estimating branch lengths based on the inferred number of substitutions, and then converting these branch lengths into units of time under the assumption of a molecular clock~\citep{zuckerkandl1965,drummond2006relaxed}.
However, phylogenetic estimates of the divergence times of many viral lineages are clearly too recent~\citep{duchene2014analyses, ho2015time, aiewsakun2016time}.
For example, the integration of filoviruses into their host genomes indicate that Ebola and Marburg virus diverged from their common ancestor 7 to 12 million years ago---but the estimate of this divergence time based on phylogenetic analyses of the viral sequences is only $\sim$10,000 years ago~\citep{carroll2013molecular, taylor2014evidence}. 
Similarly, the phylogenetic estimate of when major simian immunodeficiency virus groups diverged is almost 100 times more recent than the estimate based on the geographic isolation of their host species~\citep{wertheim2009dating, worobey2010island}. 
These examples, along with other similar discrepancies with measles virus~\citep{furuse2010origin}, coronavirus~\citep{wertheim2013case}, and hepatitis B virus~\citep{fares2002revised, holmes2003molecular}, indicate that phylogenetic methods have a systematic bias toward underestimation of deep branches.

This underestimation occurs in part because phylogenetic models do a poor job of describing the real natural selection on protein-coding genes. 
These genes evolve under purifying selection to maintain the structure and function of the proteins they encode. 
In general, these constraints are highly idiosyncratic among sites~\citep{echave2016causes}.
However, most phylogenetic models try to account for these constraints using relatively simple approaches such as allowing the rate of substitution to vary across sites according to some statistical distribution~\citep{yang1994maximum,yang2000codon}.
These models of purifying selection are usually inadequate~\citep{duchene2015substitution, duchene2015evaluating}, potentially causing branch lengths to be severely underestimated~\citep{wertheim2011purifying, halpern1998evolutionary}.

More recent work has used mutation-selection models to better account for purifying selection~\citep{halpern1998evolutionary,yang2008mutation,rodrigue2010mutation,tamuri2012estimating,mccandlish2014modeling}.
These models explicitly incorporate the fact that different protein sites prefer different amino acids, and so can improve phylogenetic estimates when there are deep branches~\citep{philippe1998good, lartillot2007suppression, le2008phylogenetic, si2008empirical, wang2008class, susko2018accelerated}.
However, these approaches require inferring the site-specific purifying selection from natural sequence data.

Even more recently, it has become possible to directly measure purifying selection on proteins using deep mutational scanning.
This high-throughput approach involves experimentally measuring how each amino-acid mutation affects protein function in the lab~\citep{fowler2014deep}.
The resulting experimental measurements of which amino acids are preferred at each protein site can be used to inform phylogenetic substitution models~\citep{bloom2014experimentally}.
These experimentally informed codon models (ExpCMs) generally exhibit much better phylogenetic fit than standard substitution models~\citep{doud2015site, hilton2017phydms, haddox2018mapping, lee2018deep}.

Here we examine how ExpCMs and other models of purifying selection estimate branch lengths on a phylogenetic tree of influenza virus hemagglutinin (HA). 
We find that ExpCMs estimate longer deep branches, and show that this extension of branch length is mostly independent and additive with that achieved by the more conventional approach of modeling rate variation. 
We also show that ExpCMs estimate similar branch lengths to a mutation-selection model that infers the amino-acid preferences from the natural sequence data rather than using values obtained in experiments.
However, all of these mutation-selection models are limited by their failure to account for another feature of purifying selection: the fact that a site's amino-acid preferences shift over time due to epistasis.
Therefore, truly accurate analyses of deep phylogenies need to account for the fact that amino-acid preferences vary across time as well as across sites.

\section*{Results}

\subsection*{Different ways substitution models account for purifying selection}
Here we consider how purifying selection is handled by codon models, which are the most sophisticated of the three classes (nucleotide, codon, and amino acid) of phylogenetic substitution models in widespread use for protein-coding genes~\citep{arenas2015trends}.
Standard codon models distinguish between two types of substitutions: synonymous and nonsynonymous.
The relative rate of these substitutions is referred to as dN/dS or $\omega$.
In their simplest form, codon substitution models fit a single $\omega$ that represents the gene-wide average fixation rate of nonsynonymous mutations relative to synonymous ones.
Here we will use such substitution models in the form proposed by \citet{goldman1994codon}.
When these models have a single gene-wide $\omega$ they are classified as M0 by \citet{yang2000codon}.
We will refer to M0 Goldman-Yang models simply as GY94 models (\ref{eq:GY94}). 
The gene-wide $\omega$ is usually $<1$~\citep{murrell2015gene}, and crudely represents the fact that many amino-acid substitutions are under purifying selection.

A single gene-wide $\omega$ ignores the fact that purifying selection is heterogeneous across sites.
The most common strategy to ameliorate this defect is to allow $\omega$ to vary among sites according to some statistical distribution~\citep{yang1994maximum,yang2000codon}.
For instance, in the M5 variant of the GY94 model~\citep{yang2000codon}, $\omega$ follows a gamma distribution as shown in \ref{fig:model_feature}A.
We will denote this model as GY94+$\Gamma\omega$.
A GY94+$\Gamma\omega$ captures the fact that the rate of nonsynonymous substitution can vary across sites. 
However, these models do not capture the fact that the same amino-acid mutation can have very different effects at different sites. 

Mutation-selection models account for the fact that purifying selection depends idiosyncratically on the specific amino-acid mutation at each site~\citep{halpern1998evolutionary,yang2008mutation,rodrigue2010mutation,tamuri2012estimating,mccandlish2014modeling}.
Here we will consider mutation-selection models where the site-specific selection is assumed to act solely at the protein level (different codons for the same amino acid are treated as selectively equivalent).
Such models explicitly define a different set of amino-acid preferences at each site in the protein. 
This more mechanistic formulation results in a site-specific stationary state (\ref{fig:model_feature}B). 
These models capture the site-to-site variation in amino-acid composition that is an obvious features of real proteins, and usually better describe actual evolution than models with only rate variation as assessed by Bayesian or maximum-likelihood criteria~\citep{lartillot2004bayesian, le2008phylogenetic, si2008empirical, wang2008class, rodrigue2010mutation,bloom2014experimentally,bloom2014informed,hilton2017phydms}.

However, the increased realism of mutation-selection models comes at the cost of an increased number of parameters. 
Codon substitution models with uniform stationary states have only a modest number of parameters that must be fit from the phylogenetic data.
For instance, a GY94+$\Gamma\omega$ model with the commonly used F3X4 stationary state has 12 parameters: two describing the shape of the gamma distribution over $\omega$, a transition-transversion rate, and nine parameters describing the nucleotide composition of the stationary state.
However, mutation-selection models must additionally specify 19 parameters defining the amino-acid preferences for \emph{each} site (there are 20 amino acids whose preferences are constrained to sum to one).
This corresponds to $19\times L$ parameters for a protein of length $L$, or 9,500 parameters for a 500-residue protein.
It is challenging to obtain values for these amino-acid preference parameters in a maximum-likelihood framework without overfitting the data~\citep{rodrigue2013statistical}.
Here we will primarily use experimentally informed codon models (ExpCMs), which define the site-specific amino-acid preference parameters \textit{a priori} from deep mutational scanning experiments so that they do not need to be fit from phylogenetic data~\citep[see Methods and][]{bloom2014experimentally, hilton2017phydms, bloom2017identification}.
Because the amino-acid preference parameters in an ExpCM are obtained from experiments, the number of ExpCM free parameters is similar to a non-site-specific substitution model.
An alternative strategy to account for site-specific amino-acid preferences is to formally consider them as random effects across sites, rather than parameters, and infer them using Bayesian methods~\citep{lartillot2004bayesian, rodrigue2014site}.
This strategy is discussed in the last section of the Results.

Importantly, these two strategies for modeling purifying selection are not mutually exclusive.
Mutation-selection models such as an ExpCM can still incorporate an $\omega$ parameter, which now represents the relative rate of nonsynonymous to synonymous substitution \emph{after} accounting for the constraints due to the site-specific amino-acid preferences~\citep{bloom2017identification,rodrigue2017detecting}.
This $\omega$ parameter for an ExpCM can be drawn from a statistical distribution (e.g., a gamma distribution) just like for GY94-style models~\citep{rodrigue2014site,haddox2018mapping}. 
We will denote such models as ExpCM+$\Gamma\omega$.
\ref{fig:model_feature}C shows the full spectrum of models that incorporate all combinations of gamma-distributed $\omega$ and site-specific stationary states.

\subsection*{Effect of stationary state and rate variation on branch-length estimation}
Given a single branch, a substitution model transforms sequence divergence into branch length.
Under a molecular-clock assumption, this branch length is proportional to time.
The transformation from sequence divergence to branch length is trivial when the sequence identity is high.
For instance, when there has only been one substitution, then the sequence identity will simply be $\frac{L - 1}{L}$ for a gene of $L$ sites, and even a simple exponential model~\citep{zuckerkandl1965} will correctly infer the short branch length of $1/L$ substitutions per site.
However, as substitutions accumulate it becomes progressively more likely for multiple changes to occur at the same site.
In this regime, the accuracy of the substitution model becomes critical for transforming sequence divergence into branch length.
Any time-homogenous substitution model predicts that after a very large number of substitutions, two related sequences will approach some asymptotic amino-acid sequence identity.
For instance, if all 20 amino acids are equally likely in the stationary state, then this asymptotic sequence identity will be $\frac{1}{20} = 0.05$.
If the substitution model underestimates the asymptotic sequence identity then it will also underestimate long branch lengths, since it will predict that sequences that have evolved for a very long time should be more diverged than is actually the case.

\ref{fig:decay} shows how different substitution models predict amino-acid sequence identity to decrease as a function of branch length using model parameters fit to a phylogeny of H1 influenza hemagglutinin (HA) genes.
The GY94 model predicts the same behavior for all sites, since it does not have any site-specific parameters, with an asymptotic sequence identity of 0.062. 
While this predicted sequence identity is higher than $\frac{1}{20} = 0.05$ due to redundant codon and nucleotide biases favoring certain amino acids, it is much lower than the pairwise identity of even the most diverged HAs in nature.
While it is of course possible that the identity of HAs in nature would become even lower given more time, it seems biochemically improbable that it would ever become as low as 0.062. 
The reason is that like many proteins HA has a highly conserved structure and function that imposes constraints that cause many sites to sample only a small subset of the 20 amino acids among all known HA homologs~\citep{nobusawa1991comparison}.

Accounting for site-to-site dN/dS rate variation in GY94 models affects the rate at which the asymptotic sequence identity is approached, but not the actual value of this asymptote.
For instance, \ref{fig:decay} shows that the GY94+$\Gamma\omega$ model takes longer to reach the asymptote than GY94, but that the asymptote is identical for both models. 
This fact holds true even if we use experimental measurements of HA's site-specific amino-acid preferences~\citep{doud2016accurate} to calculate a different $\omega_r$ value for each site using the method of \citet{spielman2015relationship} (see \ref{eq:w_r}).
Specifically, this GY94+$\omega_r$ model predicts that different sites will approach the asymptote at different rates, but the asymptote is always the same (\ref{fig:decay}).
The invariance of the asymptotic sequence identity under different schemes for modeling $\omega$ is a fundamental feature of the mathematics of this type of reversible substitution model.
These models are reversible stochastic matrices, which can be decomposed into stationary states and symmetric exchangeability matrices~\citep{nielsen2006statistical}.
The stationary state is invariant with respect to multiplication of the symmetric exchangeability matrix by any non-zero number.
Different schemes for modeling $\omega$ only multiply elements of the symmetric exchangeability matrix.
Therefore, no matter how ``well" a model accounts for site-to-site variation in $\omega$, it will always have the same stationary state as a simple GY94 model. 

However, mutation-selection models such as ExpCMs have site-specific stationary states.
They predict that different sites will have different asymptotic sequence identities (\ref{fig:decay})---a prediction that accords with the empirical observation that some sites are much more variable than others in alignments of highly diverged sequences.
For instance, \ref{fig:decay} shows that at sites such as 183 and 305 in the H1 HA, an ExpCM but not a GY94-style model predicts that the identity will always be relatively high. 
When sites with highly constrained amino-acid preferences such as these are common, an ExpCM can estimate a long branch length at modest sequence identities that a GY94 model might attribute to a shorter branch.


\subsection*{Simulations demonstrate how failure to model site-specific amino-acid preferences leads to branch-length underestimation.}

To directly demonstrate the effect of stationary state and $\Gamma\omega$ rate variation on branch-length estimation, we tested the ability of a variety of models to accurately infer branch lengths on simulated data (\ref{fig:simulations}).
Specifically, we simulated alignments of sequences along the HA phylogenetic tree using an ExpCM parameterized by the amino-acid preferences of H1 HA as experimentally measured by deep mutational scanning~\citep{doud2016accurate}. We then estimated the branch lengths from the simulated sequences using all the substitution models in \ref{fig:model_feature}C, and compared these estimates to the actual branch lengths used in the simulations.
Note that these simulations closely parallel those performed by \citet{halpern1998evolutionary} and \citet{wertheim2011purifying}.

The models with a uniform stationary state underestimated the lengths of long branches on the phylogenetic tree of the simulated sequences (\ref{fig:simulations}). 
The GY94 model estimated branch lengths that are $\sim$60\% of the true values for the longest branches. 
Accounting for site-to-site variation in $\omega$ did not fix the fundamental problem: the GY94+$\Gamma\omega$ did slightly better, but still substantially underestimated the longest branches.
However, there was no systematic underestimation of long branches by the ExpCM and ExpCM+$\Gamma\omega$ models.
The improved performance of the ExpCMs is due to their modeling of the site-specific amino-acid preferences: if we parameterize ExpCMs by amino-acid preferences that have been averaged across HA sites (and so are no longer site-specific), then they perform no better than GY94 models (\ref{fig:simulations}).
Therefore, models with uniform stationary states underestimate the length of long branches in phylogenies of sequences that have evolved under strong site-specific amino-acid preferences.


\subsection*{Experimentally informed site-specific models estimate longer branches on real data.}

The foregoing section shows the superiority of ExpCMs to GY94 models for estimating long branches on phylogenies simulated with ExpCMs.
But how do these models perform on real data?
Real genes do evolve under functional constraint, but these constraints are almost certainly more complex than what is modeled by an ExpCM.
However, if ExpCMs do a substantially better job than GY94 models of capturing the true constraints, then we might still expect them to estimate more accurate branch lengths.

To test the models on real data, we used actual sequences of influenza HA. 
The topology of HA phylogenetic trees makes these sequences an interesting test case for branch-length estimation.
HA consists of a number of different subtypes.
Sequences within a subtype have $>$68\% amino-acid identity, but sequences in different subtypes have as little as 38\% identity.
However, HA proteins from all subtypes have a highly conserved structure that performs a highly conserved function~\citep{ha2002h5,russell2004h1}.
We used \texttt{RAxML}~\citep{stamatakis2006raxml} with a nucleotide substitution model  (GTRCAT) to infer a phylogenetic tree for 92 HA sequences drawn from 15 of the 18 subtypes (we excluded bat influenza and one other rare subtype).
For the rest of this paper, we fix the tree topology to this \texttt{RAxML}-inferred tree.
Although the nucleotide model used with \texttt{RAxML} to infer this tree topology is probably less accurate than codon models, the modular subtype structure of the HA phylogeny (the tree is clearly divided into widely separated clades) means that most of the phylogenetic uncertainty lies in the length of the long branches separating the subtypes rather than in the tree topology itself.
We note that other genes may have phylogenetic structures that are more prone to topological uncertainty. 
In such cases, the accuracy of the substitution model may be important for avoiding topological errors such as long-branch attraction~\citep{felsenstein1978cases,lartillot2007suppression}.

Deep mutational scanning has been used to measure the amino-acid preferences of all sites in two different HAs.
One scan measured the preferences of an H1 HA~\citep{doud2016accurate} and the other measured the preferences of an H3 HA~\citep{lee2018deep}.
The amino-acid preferences measured for these two HAs are shown in \ref{suppfig:prefs_doud} and 
\ref{suppfig:prefs_lee}.
The H1 and H3 HAs have only $\sim$42\% amino-acid identity.
As described in \citet{lee2018deep}, the amino-acid preferences clearly differ between the H1 and H3 HA at a substantial number of sites (these differences are apparent in a simple visual comparison of \ref{suppfig:prefs_doud} and 
\ref{suppfig:prefs_lee}; see site 33 as an example).
Therefore, we also created a third set of amino-acid preferences by averaging the measurements for the H1 and H3 HAs, under the conjecture that these averaged preferences might better describe the ``average'' constraint on sites across the full HA tree~(\ref{suppfig:prefs_average}).
These three sets of HA amino-acid preferences define three different ExpCMs.
 
We fit the GY94 model and each of the three ExpCMs to the fixed HA tree topology estimated using \texttt{RAxML}, and also tested a version of each model with $\Gamma\omega$ rate variation.
\ref{tab:empirical_data} shows that all ExpCMs fit the actual data much better than the GY94 models.
The best fit was for the ExpCM informed by the average of the H1 and H3 deep mutational scans.
For all models, incorporating $\Gamma\omega$ rate variation improved the fit, although even ExpCMs without $\Gamma\omega$ greatly outperformed the GY94+$\Gamma\omega$ model (\ref{tab:empirical_data}).
As mentioned in the previous section, $\omega$ is generally $<1$ when a single value is fit to all sites in a gene~\citep{murrell2015gene}, and this is the case for all the models we tested (\ref{tab:empirical_data}).
However, the ExpCMs always fit an $\omega$ greater than the GY94 model, suggesting that the site-specific amino-acid preferences capture some of the purifying selection that the GY94 models can represent only via a small $\omega$.
Among the models with $\Gamma\omega$, the GY94+$\Gamma\omega$ model fits all four $\omega$ categories to values $\ll$1, but the ExpCM+$\Gamma\omega$ models fit one of the $\omega$ categories to a value $>1$.
This increase in $\omega$ values makes sense given the different interpretation of $\omega$ for each family of models. 
The ExpCM $\omega$ is the relative rate of fixation of nonsynonymous to synonymous mutations \textit{after} accounting for the functional constraints described by the amino-acid preferences.
This more realistic null model gives ExpCMs enhanced power to detect diversifying selection for amino-acid change~\citep{bloom2017identification, rodrigue2017detecting}, which is known to occur at some sites in HA due to immune selection~\citep{bedford2014integrating}.

Importantly, models that account for purifying selection via either $\Gamma\omega$ rate variation or site-specific amino-acid preferences do not just exhibit better fit---they also estimate longer branches on the HA tree. 
\ref{fig:empirical_trees} shows the branch lengths optimized by each model on a common scale.
The tree's deepest branches are shortest when they are optimized by the GY94 model, which lacks both $\Gamma\omega$ and site-specific amino-acid preferences.
Adding either $\Gamma\omega$ rate variation or site-specific amino-acid preferences increases the length of the deep branches.
Specifically, the tree's diameter (the distance between the two most divergent tips) for the GY94+$\Gamma\omega$ model is 159\% of the GY94 model tree diameter (\ref{supptab:tree_diameter}).
The tree diameter is 122\% and 135\% of the GY94 model tree diameter for ExpCMs informed by H1 or H3 amino-acid preferences, respectively, and 160\% of the GY94 model for the ExpCM informed by the average of the H1 and H3 preferences (\ref{supptab:tree_diameter}).

The deepening of branch lengths that results from the $\Gamma\omega$ and site-specific amino-acid preference approaches to modeling purifying selection are largely independent.
This can be seen by examining the ExpCM+$\Gamma\omega$ models, which combine $\Gamma\omega$ rate variation with site-specific amino-acid preferences.
As shown in \ref{fig:empirical_trees}, these ExpCM+$\Gamma\omega$ models estimate longer branches than models with just $\Gamma\omega$ rate variation (GY94+$\Gamma\omega$) or just site-specific amino-acid preferences (ExpCMs).
The near independence of these effects is quantified in \ref{supptab:tree_diameter}, which shows that 76\% of the tree diameter extension of ExpCM(H1+H3 avg)+$\Gamma\omega$ versus can be explained by simply adding the extension from incorporating $\Gamma\omega$ (GY94+$\Gamma\omega$ versus GY94) to the extension from incorporating site-specific amino-acid preferences (compare ExpCM(H1+H3 avg) to GY94).

However, while adding $\Gamma\omega$ rate variation increases the length of deep branches in a roughly uniform fashion across the tree, the branch lengthening from adding site-specific amino-acid preferences is not uniform across the tree (\ref{fig:empirical_trees} and \ref{fig:empirical_scatter}). 
Instead, the increase in branch length is most pronounced on branches leading to the HA sequence that was used in the deep mutational scanning experiment that informed the ExpCM.
For instance, the ExpCM informed by the H1 data most dramatically lengthens branches near the H1 clade of the tree, while the ExpCM informed by the H3 data has the largest effect on branches near the H3 clade.
The ExpCM informed by the average of the H1 and H3 data has a more uniform effect across the tree, but still most strongly extends branches leading to either the H1 or H3 clade.
Therefore, \ref{fig:empirical_trees} and \ref{fig:empirical_scatter} show that ExpCMs estimate longer branches, but that the effect is shaped by the set of amino-acid preferences used to inform the model.

\subsection*{Shifting amino-acid preferences limit the benefits of models with site-specific stationary states for estimating long branch lengths.}

The fact that an ExpCM leads to the most profound increase in branch length leading to the sequence used in the experiment can be rationalized in terms of existing knowledge about epistasis during protein evolution.
Each ExpCM is informed by a single set of experimentally measured amino-acid preferences.
But in reality, the effect of a mutation at one site in a protein can depend on the amino-acid identities of other sites in the protein~\citep{ortlund2007crystal, gong2013stability, harms2014historical, tufts2014epistasis, starr2018pervasive}. 
This epistasis can lead to shifts in a protein's amino-acid preferences over evolutionary time~\citep{pollock2012amino, doud2015site, shah2015contingency, bazykin2015changing, haddox2018mapping}.
Because the deep mutational scanning experiments that inform our ExpCMs were each performed in the context of a single HA genetic background, their measurements do not account for the accumulation of epistatic shifts in amino-acid preferences as HA evolves. 
Therefore, an ExpCM is expected to most accurately describe the evolution of sequences closely related to the one used in the experiment.
 
We can observe how shifting amino-acid preferences degrade the accuracy of an ExpCM by fitting the model to trees containing increasingly diverged sequences.
For both H1 and H3 HAs, we created three phylogenetic trees (\ref{suppfig:subalignments}): a ``low'' divergence tree that contains sequences with $\ge$59\% amino-acid identity to the HA used in the experiment, an ``intermediate'' divergence tree that contains sequences with $\ge$46\% amino-acid identity to the HA in the experiment, and a ``high'' divergence tree that contains all HAs (which have as little as 38\% identity to the HA in the experiment).
\ref{fig:compete} shows the subtrees containing each of these sets of HA sequences.
For each subtree, we examined the congruence between site-specific natural selection and the amino-acid preferences measured in the deep mutational scanning experiment using the ExpCM stringency parameter $\beta$~\citep{bloom2014informed,hilton2017phydms}. 
Values of $\beta$ that are $>$1 indicate that natural selection prefers the same amino acids as the experiments but with a greater stringency, suggesting strong congruence between natural selection and the experimental preferences. 
In contrast, values of $\beta$ that are $<$1 flatten the preferences, suggesting that they provide a relatively poor description of natural selection on the protein.

The value of $\beta$ decreases as the divergence from the sequence used in the deep mutational scan increases~\ref{fig:compete}. 
This inverse relationship between $\beta$ and overall divergence is seen for the ExpCMs informed by both the H1 and H3 experiments.
As $\beta$ decreases, the preferences ``flatten" and so the ExpCM draws less information from the experiment. 
At the most extreme value of $\beta = 0$, the preferences would be perfectly uniform and look similar to the GY94 preferences in \ref{fig:empirical_trees}.
In reality, $\beta$ never reaches a value this low, indicating the deep mutational scanning experiments remain somewhat informative about real natural selection across the entire swath of HAs. 
However, \ref{fig:compete} shows that the amino-acid preferences clearly become less informative about natural selection as we move away from the experimental sequence on the tree.
This shifting of amino-acid preferences helps explain why the ExpCM informed by the average of the H1 and H3 experiments performs best (\ref{tab:empirical_data}, \ref{fig:empirical_trees}, and \ref{fig:empirical_scatter}): averaging the measurements across these two HAs is a heuristic method of accounting for shifts in preferences during HA evolution. 
 
The fact that amino-acid preferences shift as a protein evolves leaves us with an inherent tension: models with site-specific amino-acid preferences only become important for accurate branch-length estimation as sequences become increasingly diverged, but this same divergence degrades the accuracy of extrapolating the amino-acid preferences from any given experiment across the phylogenetic tree.
Crucially, this problem is more fundamental than the inability of a single deep mutational scanning experiment to measure amino-acid preferences in more than one genetic background.
If amino-acid preferences shift during evolution, there simply will not be any single set of time-homogeneous site-specific preferences that accurately describes evolution along the entirety of a phylogenetic tree that covers a wide span of sequences.

\subsection*{A model with amino-acid preferences estimated from natural sequences gives similar results to an ExpCM}

The previous sections used ExpCMs, which are mutation-selection models that use site-specific amino-acid preferences that have been measured by experiments. 
However, there are other mathematically similar implementations of mutation-selection models that infer the amino-acid preferences directly from the natural sequence data. 
When these models are designed for use in phylogenetic inference, they are generally implemented in a Bayesian framework, which avoids the overfitting problems associated with trying to make maximum-likelihood estimates of the thousands of amino-acid preference parameters~\citep{lartillot2014overcoming}.
(Note that the maximum-likelihood implementations of \citet{tamuri2012estimating,tamuri2014penalized} are designed for estimating the amino-acid preferences, \emph{not} for phylogenetic inference.)
The model most comparable to our ExpCMs is the codon mutation-selection model implemented in \texttt{PhyloBayes-MPI}, which we will refer to as pbMutSel~\citep{rodrigue2014site}. 
In the pbMutSel model, the amino-acid preferences are modeled using Dirichlet processes rather than derived from experiments. 
However, like an ExpCM, a pbMutSel model still assumes a single set of time-homogeneous site-specific amino-acid preferences for the entire tree.

Comparing ExpCM and pbMutSel models can help determine the ultimate limits of mutation-selection models that assign each site a single set of amino-acid preferences. 
If the limitations of ExpCMs described above arise simply because the deep mutational scanning experiments do not correctly measure the ``true" amino-acid preferences across the entirety of a highly diverged phylogenetic tree, then we would expect the pbMutSel models (which infer these preferences from the entire tree) to perform better.
On the other hand, if the major limitation is that no single set of time-homogenous amino-acid preferences can fully describe evolution over the entire tree, then we would expect ExpCM and pbMutSel models to perform similarly.

We fit a pbMutSel model to the entire HA phylogenetic tree, and compared the results to those from analyzing the same tree with the best ExpCM, which is the ExpCM(H1+H3 avg)+$\Gamma\omega$ variant.
This is a direct apple-to-apples comparison, since the pbMutSel model also draws $\omega$ from a gamma-distribution~\citep{rodrigue2014site}.
First, we compared the amino-acid preferences inferred by the pbMutSel model to the preferences measured in the experiments.
\ref{fig:phylobayes}A shows that the preferences inferred by pbMutSel are quite similar to the (H1+ H3 avg) obtained by averaging the deep mutational scanning measurements for the H1 and H3 HAs. 
Notably, the amino-acid preferences from the pbMutSel model are more correlated with the (H1+ H3 avg) than the H1 and H3 measurements are with each other (\ref{fig:phylobayes}A).
This strong correlation indicates that the ExpCM(H1+H3 avg)+$\Gamma\omega$ is unlikely to be much different than a pbMutSel model that is parameterized only using the natural sequence data.  

We next compared the branch lengths estimated by using the ExpCM(H1+H3 avg)+$\Gamma\omega$ and pbMutSel models.
As shown in \ref{fig:phylobayes}B, these two models estimated similar branch lengths across the entire HA phylogenetic tree. 
However, the estimates are not identical, and the tension between local and global accuracy of the amino-acid preferences is still apparent. 
Specifically, the branches leading to the H1 or H3 sequences used in the experiments were estimated to be slightly longer by the ExpCM, while some other branches were estimated to be slightly longer by the pbMutSel model. 
The relatively longer branches leading to the experimental sequences when using the ExpCM(H1+H3 avg)+$\Gamma\omega$ suggests that the ``tree average" amino-acid preferences inferred by the pbMutSel model are not as accurate as the preferences from the deep mutational scanning for sequences close to those used in the experiments. 
However, for sequences distant from those used in the experiments, the ``tree average'' preferences inferred by the pbMutSel model appear to be slightly better than the experimental values.
Therefore, while the ExpCM and pbMutSel models differ slightly in the extent to which they lengthen different branches, neither model can avoid the tension between the local and global accuracy of amino-acid preferences. 

\section*{Discussion}
 
We examined how estimates of deep branch lengths on phylogenetic trees are affected by accounting for the fact that proteins prefer specific amino acids at specific sites. 
We did this by comparing inferences from models informed by experimental measurements of site-specific amino-acid preferences with more conventional codon substitution models, as well as with models that infer the amino-acid preferences from the natural sequences.
We found that models that account for site-specific amino-acid preferences estimated deeper long branches, regardless of whether these preferences are measured experimentally or inferred from the sequence alignment.
Additionally, we showed that the extension in branch length from site-specific amino-acid preferences is mostly independent of the extension that results from simply modeling rate variation. 

Overall, our results underscore the importance of modeling purifying selection in a way that is more nuanced than simply allowing the substitution rate to vary across sites.
Protein sites do not simply differ in their rates of substitution---different sites also prefer different amino acids.
There are now two ways to account for this fact: using models informed by deep mutational scanning experiments, or using models that infer site-specific amino-acid preferences from the natural sequence alignment.
Combining either type of model with rate variation increases the inferred length of deep branches relative to models that only incorporate rate variation.
We expect that further improvements could be achieved by also incorporating other factors such as host-specific substitution rates~\citep{worobey2014synchronized} that are known to be important for modeling the evolution of viral genes such as HA.

However, assuming a single set of site-specific amino-acid preferences is still an imperfect way to model evolution over a highly diverged phylogenetic tree. 
In the case of the experimentally informed models, it is fairly obvious why this is true: the amino-acid preferences are measured in just one genetic background, and therefore provide only a single snapshot of preferences that shift over evolutionary time due to epistasis~\citep{pollock2012amino, shah2015contingency, bazykin2015changing, haddox2018mapping, starr2018pervasive, doud2015site}. 
As a result, experimentally measured amino-acid preferences are most accurate for sequences similar to the one used in the experiment, and so cause the largest increases in branch length in that region of the phylogenetic tree.
However, this limitation is not unique to experimentally informed models, but is a general limitation of describing purifying selection using a single set of site-independent and time-homogenous amino-acid preferences.
For instance, we showed that averaging experimental measurements on two protein homologs does a somewhat better job of capturing the ``average'' constraint across the tree, and performs similarly to approaches that infer the ``average'' preferences from natural sequence data~\citep{rodrigue2010mutation,rodrigue2014site}.
But even these ``average'' preferences exhibit a tradeoff between local and global accuracy for the inference of deep branch lengths.

So while modeling site-specific amino-acid preferences is a clear improvement over most conventional models, the next step towards greater accuracy will require relaxing the assumption that these preferences are time homogeneous and site independent.
Of course, many authors have pointed out the shortcomings of models that fail to account for the full site-interdependent complexity of purifying selection~\citep{rodrigue2005site, choi2007quantifying, pollock2012amino, goldstein2017sequence}.
However, the challenge is to overcome these shortcomings with models that are tractable for real phylogenetic questions.
There are two main issues: first, the Felsenstein pruning algorithm~\citep{felsenstein1981evolutionary} that is typically used to evaluate phylogenetic likelihoods breaks down when sites are no longer treated independently.
Some alternative algorithms have been proposed~\citep{bordner2013new,rodrigue2009computational,rodrigue2005site, choi2007quantifying}, but they are still in their infancy.
Second, site-interdependent models require a realistic ``fitness function'' that describes the interactions among sites.
It appears that typical structural modeling programs are insufficient for this purpose~\citep{rodrigue2009computational}.
But hope comes from experimental progress in measuring actual site-interdependent constraints on proteins~\citep{olson2014comprehensive,wu2016adaptation,steinberg2016shifting,li2016fitness}, combined with new methods for using these measurements to parameterize fitness functions~\citep{sailer2017detecting,otwinowski2018inferring}.
Perhaps some day such truly realistic models might be useful for phylogenetic inference.
Until that day, our work shows that modeling a single set of time homogenous amino-acid preferences provides at least some improvement.

\clearpage
\section*{Methods}

\subsection*{Substitution models}
All of the substitution models used in this paper have been described previously.
However, here we briefly recap their exact mathematical implementations.

\subsubsection*{GY94 model}

The GY94 model is M0 variant of the Goldman-Yang model described by \citet{yang2000codon}. 
Specifically, the substitution rate $P_{xy}$ from codon $x$ to codon $y$ is 
\begin{equation}
\label{eq:GY94}
P_{xy} = 
\begin{cases}
  0 & \mbox{if $x$ and $y$ differ by more than one nucleotide,}\\
  \Phi_y & \mbox{if $\mathcal{A}\left(x\right) = \mathcal{A}\left(y\right)$ and $x$ is converted to $y$ by a single-nucleotide transversion,} \\
  \omega \Phi_{y} & \mbox{if $\mathcal{A}\left(x\right) \ne \mathcal{A}\left(y\right)$ and $x$ is converted to $y$ by a single-nucleotide transversion,} \\
  \kappa \Phi_y & \mbox{if $\mathcal{A}\left(x\right) = \mathcal{A}\left(y\right)$ and $x$ is converted to $y$ by a single-nucleotide transition,} \\
  \omega \kappa \Phi_{y} & \mbox{if $\mathcal{A}\left(x\right) \ne \mathcal{A}\left(y\right)$ and $x$ is converted to $y$ by a single-nucleotide transition,} \\
  -\sum\limits_{z \ne x} P_{xz} & \mbox{if $x = y$,}
  \end{cases}
\end{equation}
where $\mathcal{A}\left(x\right)$ is the amino-acid encoded by codon $x$, $\kappa$ is the transition-transversion rate, $\Phi_y$ is the equilibrium frequency of codon $y$, and $\omega$ is the relative rate of nonsynonymous and synonymous substitutions.
We define the codon frequency parameters, $\Phi_y$, using the ``corrected F3X4" method from~\citet{pond2010correcting}. 
There are nine parameters describing the nucleotide frequencies at each codon site (the nucleotides are constrained to sum to one at each codon position),  and these parameter values are calculated from the empirical alignment frequencies. 
The ``corrected F3X4" method calculates the $\Phi_y$ values from these nucleotide frequencies but corrects for the exclusion of sequences with premature stop codons from the analysis. 

The frequency $p_x$ of codon $x$ in the stationary state of a GY94 model is simply 
\begin{equation}
\label{eq:px}
p_{x} = \Phi_x.
\end{equation}
Overall, a GY94 model has 11 free parameters: $\kappa$, $\omega$, and the 9 nucleotide frequency parameters used to define $\Phi_y$. 

\subsubsection*{Experimentally Informed Codon Model (ExpCM)}

The ExpCM models used in this paper are the ones described in \citet{bloom2017identification}. 
Briefly, the rate of substitution $P_{r,xy}$ of site $r$ from codon $x$ to $y$ is 
\begin{equation}
\label{eq:ExpCM}
P_{r,xy} = Q_{xy} \times F_{r,xy}
\end{equation}
where $Q_{xy}$ is proportional to the rate of mutation from $x$ to $y$, $F_{r,xy}$ is proportional to the probability that this mutation fixes, and the diagonal elements $P_{xx}$ are set by $P_{xx} = -\sum_{z \ne x} P_{xz}$. 

The rate of mutation $Q_{xy}$ is assumed to be uniform across sites, and takes an HKY85-like~\citep{hasegawa1985dating} form as 
\begin{equation}
\label{eq:Qxy}
   Q_{xy} = \begin{cases}
            0 & \mbox{if $x$ and $y$ differ by more than one nucleotide,} \\
            \phi_w & \mbox{if $x$ can be converted to $y$ by a transversion of a nucleotide to $w$,} \\
            \kappa \times \phi_w & \mbox{if $x$ can be converted to $y$ by a transition of a nucleotide to $w$} \\
            \end{cases}
\end{equation}
where $\phi_w$ is the nucleotide frequency of nucleotide $w$ and $\kappa$ is the transition-transversion rate.


The deep mutational scanning amino-acid preferences are incorporated into the ExpCM via the $F_{r,xy}$ terms.
The experiments measure the preference $\pi_{r,a}$ of every site $r$ for every amino-acid $a$.
$F_{r,xy}$ is defined in terms of these experimentally measured amino-acid preferences as
\begin{equation}
\label{eq:Frxy}
F_{r,xy} = 
\begin{cases}
   1 & \mbox{if $\mathcal{A}\left(x\right) = \mathcal{A}\left(y\right)$,} \\
   \omega \times \frac{\ln\left[\left(\pi_{r,\mathcal{A}\left(y\right)} / \pi_{r,\mathcal{A}\left(x\right)}\right)^{\beta}\right]}{1 - \left(\pi_{r,\mathcal{A}\left(x\right)} / \pi_{r,\mathcal{A}\left(y\right)}\right)^{\beta}} & \mbox{if $\mathcal{A}\left(x\right) \ne \mathcal{A}\left(y\right)$,}
   \end{cases}
\end{equation}
where $\beta$ is the stringency parameter~\citep{bloom2014informed,hilton2017phydms} and $\omega$ is the relative rate of nonsynonymous to synonymous substitutions after accounting for the amino-acid preferences.

The stationary state of an ExpCM is 
\begin{equation}
\label{eq:prx}
p_{r,x} = \frac{\phi_{x_1}\phi_{x_2}\phi_{x_3}\left(\pi_{r,A\left(x\right)}\right)^\beta}{\sum_z \phi_{z_1}\phi_{z_2}\phi_{z_3}\left(\pi_{r,A\left(z\right)}\right)^\beta}
\end{equation}
where $\phi_{x_1}$, $\phi_{x_2}$, and $\phi_{x_3}$ are the nucleotides at position 1, 2, and 3 of codon $x$. 

An ExpCM has five free parameters: $\kappa$, $\omega$, and the three independent $\phi_x$ values.
The amino-acid preferences $\pi_{r,a}$ are \emph{not} free parameters since they are determined \textit{a priori} by an experiment independent of the sequence alignment being analyzed.

\subsubsection*{$\Gamma\omega$ rate variation}

The GY94+$\Gamma\omega$ is equivalent to the M5 model in \citet{yang2000codon} with $\omega$ drawn from $K = 4$ categories.
The ExpCM+$\Gamma\omega$ similarly draws $\omega$ from a $\Gamma$ distribution discretized into $K=4$ bins. 
Each bin is equally weighted and $\omega$ takes on the mean value of the bin. 
Because the $\Gamma$ distribution is defined by two parameters, adding $\Gamma\omega$ to a model with a single $\omega$ adds one free parameter.
Therefore, the GY94+$\Gamma\omega$ model has 12 free parameters, and the ExpCM+$\Gamma\omega$ model has 6 free parameters.

\subsubsection*{GY94 with $\omega_r$}
In \ref{fig:decay}, we describe GY94 models where each site $r$ has its own $\omega_r$ value that is calculated from the amino-acid preferences using the relationship described by \citet{spielman2015relationship}.
This relationship defines the expected rate of nonsynonymous to synonymous substitutions given the amino-acid preferences.
We first fit an ExpCM to the ``low divergence" H1 subtree (parameter values in \ref{supptab:fit_params}), which allows us to calculate $P_{r,xy}$ (\ref{eq:ExpCM}), $Q_{xy}$ (\ref{eq:Qxy}), and $p_{r,x}$ (\ref{eq:prx}).
We then calculated $\omega_r$ using the equation of \citet{spielman2015relationship}, normalizing by the gene-wide $\omega$ fit by the ExpCM:
\begin{equation}
\label{eq:w_r}
\omega_{r} = \frac{\sum_{x} \sum_{y \in N_x} {p_{r,x} \times \frac{P_{r,xy}}{\omega}}}{\sum_{x} \sum_{y \in N_x} {p_{r,x} \times Q_{xy}}},
\end{equation}
where $N_x$ is the set of codons that are nonsynonymous to codon $x$ and differ from codon $x$ by only one nucleotide. 

\subsection*{HA amino-acid preferences from deep mutational scanning experiments}
We used amino-acid preferences measured in deep mutational scans of the A/WSN/1933 H1 HA~\citep{doud2016accurate} and the A/Perth/2009 H3 HA~\citep{lee2018deep} to define the amino-acid preferences that inform the ExpCMs. 
We only used sites that can be unambiguously aligned in these H1 and H3 HAs. 
These alignable sites and their mapping to sequential numbering of the HA sequences used in the deep mutational scanning experiments are in \ref{suppfile:WSN_Perth_map}. 
The experimentally measured amino-acid preferences masked to just include these alignable sites are in \ref{suppfile:H1_prefs} and \ref{suppfile:H3_prefs}.
For the average preference set, we took the pairwise average of the H1 and H3 preferences. 
The preference for every amino acid $a$ at every site $r$ in the average preference set is
\begin{equation}
\label{eq:pi_avg}
\pi_{r,a,\rm{(H1+H3\ avg)}} = \frac{\pi_{r,a,\rm{H1}} + \pi_{r,a,\rm{H3}}}{2}
\end{equation}

\subsection*{HA sequences and tree topology}

We downloaded all full-length, coding sequences for 15 of the 18 influenza A virus HA subtypes from the Influenza Virus Resource Database~\citep{bao2008influenza} in June of 2017. 
We excluded rare subtypes 15, 17, and 18, which have few sequences in the database.  
We filtered and aligned the sequences using \texttt{phydms\_prepalignment}~\citep{hilton2017phydms}. 
Specifically, we used \texttt{phydms\_prepalignment} with the flag \texttt{--minidentity 0.3} to remove sequences with ambiguous nucleotides, premature stops, or frameshift mutations as well as redundant sequences.  
We also removed all codon sites which that are not alignable between the H1 HA and H3 HA used in the deep mutational scanning experiments (these alignable sites are listed in  \ref{suppfile:WSN_Perth_map}). 
We subsampled the remaining sequences to five per subtype with $\le 1$ sequence per year per subtype. 
We also included a small number of sequences from the major human and equine influenza lineages to ensure representation of these well-studied lineages. 
The resulting alignment contains 92 sequences, and is provided in \ref{suppfile:alignment_high}. 

We created four subalignments with ``low" and ``intermediate" divergence from either the H1 or the H3 deep mutational scanning reference sequence for the analysis in \ref{fig:compete}. 
The ``low divergence" alignments had $\ge 59\%$ amino-acid identity to the sequence used in the deep mutational scanning, and the ``intermediate divergence" alignments had $\ge 46\%$ identity from the reference sequence~(\ref{suppfig:subalignments}).

We inferred the tree topology of each alignment using \texttt{RAxML}~\citep{stamatakis2006raxml} and the GTRCAT model. 
We estimated the branch lengths of this fixed topology using each ExpCM and GY94 models with \texttt{phydms\_comprehensive}~\citep{hilton2017phydms}. 

\subsection*{Asymptotic amino-acid sequence identity}
For the analysis in \ref{fig:decay}, we fit models to the ``low divergence" H1 subtree.
This gave the parameter values in \ref{supptab:fit_params}. 

For each model, we calculated the expected amino-acid sequence identity for two sequences separated by a branch length of $t$ as 
\begin{equation}
\label{eq:f}
\sum_a \sum_{x \in a} p_{r,x} \sum_{y \in a} \left[e^{t\mathbf{P_r}}\right]_{xy}
\end{equation}
where $a$ ranges over all 20 amino acids, $x \in a$ indicates that $x$ ranges over all codons that encode amino-acid $a$, $p_{r,x}$ is the stationary state of the model at site $r$ and codon $x$ (given by \ref{eq:px} for GY94-family models, and \ref{eq:prx} for ExpCM-family models), and $\left[e^{t\mathbf{P_r}}\right]_{xy}$ is the value in row $x$ and column $y$ of the matrix obtained by exponentiating the product of $t$ and the substitution matrix $\mathbf{P_r}$ for site $r$ (defined by \ref{eq:GY94} for GY94-family models and \ref{eq:ExpCM} for ExpCM-family models).

\subsection*{Simulations}
For \ref{fig:simulations}, we simulated sequences using \texttt{pyvolve}~\citep{spielman2015pyvolve} along the full HA tree using an ExpCM defined by parameters fit to the ``low divergence'' H1 subtree~(\ref{supptab:fit_params}). We performed 10 replicate simulations and estimated the branch lengths for each replicate using \texttt{phydms\_comprehensive}~\citep{hilton2017phydms}. 

\subsection*{pbMutSel inference with \texttt{PhyloBayes-MPI}.}
For \ref{fig:phylobayes}, we fit a pbMutSel model to the full HA tree. 
We ran one chain for 5500 steps, saved every sample, and discarded the first 550 samples as a burnin. 
We used \texttt{PhyloBayes-MPI} program \texttt{readpb\_mpi} to compute the majority-rule consensus tree and the posterior average site-specific amino-acid preferences. 
Convergence was assessed visually using Tracer \citep{rambaut2018posterior} and by the correlation of amino-acid preferences inferred by two independent chains (r=0.996). 

In order to make the branch lengths in \ref{fig:phylobayes} comparable between the pbMutSel tree returned by \texttt{PhyloBayes-MPI} and the other trees returned by \texttt{phydms}, we we normalized the branch lengths on the pbMutSel consensus tree and the ExpCM(H1+H3 avg)+$\Gamma\omega$ by dividing each branch by the length from A/South Carolina/1/1918 and A/Solomon Islands/3/2006. 
These two H1 sequences are early and late representatives of the longest known human influenza lineage, and are of sufficiently high identity that different ExpCM and GY94 substitution models all estimate nearly identical branch lengths separating them. 

\subsection*{Software versions and computer code}
 All code used for the analyses in this paper is available at \url{https://github.com/jbloomlab/divergence_timing_manuscript}.
The external computer programs that we used were
\begin{itemize} 
\item \texttt{phydms}~\citep{hilton2017phydms} version 2.2.2 (available at \url{github.com/jbloomlab/phydms}) to fit the ExpCM and GY94 models.
\item \texttt{pyvolve}~\citep{spielman2015pyvolve} version 0.8.7 (available at \url{https://github.com/sjspielman/pyvolve}) to simulate the sequences.
\item \texttt{PhyloBayes-MPI}~\citep{rodrigue2014site} version 1.8 (available at \url{https://github.com/bayesiancook/pbmpi}) to fit the pbMutSel model. 
\item \texttt{RAxML}~\citep{stamatakis2006raxml} version 8.2.11 (available at \url{https://github.com/stamatak/standard-RAxML}) to infer tree topology.
\item We used \texttt{ggplot2}~\citep{wickham2016ggplot2}, \texttt{ggtree}~\citep{yu2017ggtree}, and \texttt{ggseqlogo}~\citep{wagih2017ggseqlogo} for visualization of the results.
\item \texttt{snakemake}~\citep{koster2012snakemake} version 3.11.2 (available at \url{https://snakemake.readthedocs.io/en/stable/}) to run the pipelines.
\end{itemize}

\subsection*{Acknowledgments}
We thank Erick Matsen and Trevor Bedford for helpful comments about the project and manuscript.
SKH is supported in part by training grant T32AI083203 from the NIAID of the National Institutes of Health.
This work was supported by the NIAID and NIGMS of the NIH under grant numbers R01AI127893 and R35GM126911. 
JDB is supported in part by a Faculty Scholars grant from the Howard Hughes Medical Institute and the Simons Foundation.
The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.
All code and data used for the analyses in this paper is available at \url{https://github.com/jbloomlab/divergence_timing_manuscript}.

\clearpage 
\bibliographystyle{mbe}
{\small
\bibliography{references.bib}
}

\newpage 
%% figures 
\begin{figure}[h!]
\centerline{\includegraphics[width=0.90\textwidth]{fig1.pdf}}
\caption{\label{fig:model_feature}
\textbf{Different ways codon models account for purifying selection.}
(A) The dN/dS parameter, $\omega$, can be defined as one gene-wide average (orange triangle) or allowed to vary according to some statistical distribution (blue line). 
For computational tractability, the distribution is discretized into $K$ bins and $\omega$ takes on the mean of each bin (blue circles)~\citep{yang1994maximum,yang2000codon}. 
A gamma distribution (denoted by $\Gamma$) with $K=4$ bins is shown here.
(B) A substitution model's stationary state defines the expected sequence composition after a very long evolutionary time. 
Most substitution models have stationary states that are uniform across sites.
However, substitution models can have site-specific stationary states.
In the logo plots, each column is a site in the protein and the height of each letter is the frequency of that amino acid at stationary state. 
(C) Substitution models can incorporate neither, one, or both of these features.
Here we will use substitution models from the Goldman-Yang~\citep[GY94;][]{goldman1994codon,yang2000codon} and experimentally informed codon model~\citep[ExpCM;][]{hilton2017phydms} families with and without gamma-distributed $\omega$ to represent all possible combinations.
}
\end{figure}
\newpage
\begin{figure}[h!]
\centerline{\includegraphics[width=0.90\textwidth]{fig2.pdf}}
\caption{\label{fig:decay}
\textbf{Effect of stationary state and $\Gamma\omega$ rate variation on predicted asymptotic sequence divergence.}
The logo plots at top show the amino-acid preferences for some sites in an H1 influenza hemagglutinin protein as experimentally measured by~\citet{doud2016accurate}.
The graphs show the expected amino-acid identity at that site for two sequences separated by a branch of the indicated length (\ref{eq:f}).
For the GY94 model, the graphs are identical for all sites since this model does not have site-specific parameters; the same is true for GY94+$\Gamma\omega$.
The graphs do differ among sites if we calculate a different $\omega_r$ for each site $r$ in the GY94 model using the amino-acid preferences~\citep[\ref{eq:w_r};][]{spielman2015relationship}.
However, all GY94 models, including the one with site-specific $\omega_r$ values, approach the same asymptote since they all have the same stationary state.
The ExpCM has different asymptotes for different sites since it accounts for how amino-acid preferences lead to site-specific stationary states.
}
\end{figure}
\newpage
\begin{figure}[h!]
\centerline{\includegraphics[width=0.9\textwidth]{fig3.pdf}}
\caption{\label{fig:simulations}
\textbf{Branch lengths inferred on data simulated under a model with site-specific amino-acid preferences.} 
We simulated alignments along a HA phylogenetic tree using an ExpCM parameterized by the actual site-specific amino-acid preferences for an H1 HA~\citep{doud2016accurate}.
We then inferred the branch lengths of this tree from the simulated alignments.
The inferred branch lengths for various models are plotted on the x-axis, and the actual branch lengths used in the simulations are on the y-axis.
We performed 10 simulations and inferences, and gray points show each inferred branch length from each simulation, and black points show the average of each branch length across simulations.
The grey dashed line at $y=x$ represents what would be seen if the inferred branch lengths exactly matched those used in the simulations. 
}
\end{figure}
\newpage
\begin{figure}[h!]
\centerline{\includegraphics[width=\textwidth]{fig4.pdf}}
\caption{\label{fig:empirical_trees}
\textbf{Effect of site-specific amino-acid preferences and $\Gamma\omega$ rate variation on HA branch length estimation.} 
The branch lengths of the HA tree were optimized using the indicated ExpCM or GY94 model. 
The amino-acid preferences defining the model (ExpCM) or implied by the model (GY94) are shown as logo plots for 15 sites in HA; the full set of experimentally measured amino-acid preferences are in \ref{suppfig:prefs_doud}, \ref{suppfig:prefs_lee}, and \ref{suppfig:prefs_average}. 
The ExpCMs use amino-acid preferences measured in deep mutational scanning of an H1 HA~\citep{doud2016accurate}, an H3 HA~\citep{lee2018deep}, or the average of the measurements for these two HAs.
Circle denotes the H1 clade and triangle denotes the H3 clade.
The root of each tree is placed where it would fall if the tree was midpoint rooted using the branch lengths inferred by \texttt{RAxML} using the GTRCAT model. 
This figure enables qualitative visualization of the trees; for a quantitative comparison of branch lengths optimized by different models, see \ref{fig:empirical_scatter}. 
}
\end{figure}
\newpage
\begin{figure}[h!]
\centerline{\includegraphics[width=\textwidth]{fig5.pdf}}
\caption{\label{fig:empirical_scatter}
\textbf{Modeling site-specific amino-acid preferences using deep mutational scanning experiments extends branch lengths, especially for branches leading to the HA used in the experiment.}
The points indicate the total length of branches separating all pairs of tips on the HA phylogenetic tree when the tree is optimized under the indicated model. 
Blue and orange denote branches that lead to the H1 and H3 HAs used in the deep mutational scanning. 
The amino-acid preference set defining the ExpCM is labeled above each each plot.
(A) ExpCMs defined by amino-acid preferences from any of the deep mutational scanning experiments estimate generally longer branches than the GY94 model, with the increase particularly pronounced for branches leading to the HA used in the experiment. 
(B) The addition of $\Gamma\omega$ rate variation further extends branch lengths, without any apparent bias towards the HAs used in the experiment.
Note that this figure shows the same data as \ref{fig:empirical_trees} in a different form.
}
\end{figure}
\newpage
\begin{figure}[h!]
\centerline{\includegraphics[width=0.75\textwidth]{fig6.pdf}}
\caption{\label{fig:compete}
\textbf{The congruence between natural selection and the deep mutational scanning measurements decreases with sequence divergence.} 
We fit an ExpCM informed by the H1 or H3 deep mutational scanning experiments to trees spanning sequences with low, intermediate, and high divergence from the sequence used in the experiment. 
The ExpCM stringency parameter ($\beta$) is a measure of the congruence between natural selection and the experimental measurements~\citep{bloom2014informed,hilton2017phydms}. 
Larger values of $\beta$ indicate that natural selection prefers the same amino acids as the experiments but with greater stringency. 
As divergence increases between the HA used in the experiment and the other sequences in the tree, the $\beta$ value decreases and the amino-acid preference ``flatten."
Therefore, the preferences measured in each experiment are progressively less congruent with natural selection as we include increasingly diverged sequences. 
}
\end{figure}
\newpage
\begin{figure}[h!]
\centerline{\includegraphics[width=0.99\textwidth]{fig7.pdf}}
\caption{\label{fig:phylobayes}
\textbf{Models inferred from natural sequences have similar stationary states to models defined by experimental preferences and estimate similar branch lengths.}
We fit an ExpCM(H1+H3 avg)+$\Gamma\omega$ and a pbMutSel to the full HA tree in \ref{fig:empirical_trees}. 
The pbMutSel amino-acid preferences are inferred from the natural HA sequences, while the ExpCM amino-acid preferences are experimentally measured and then rescaled by the stringency parameter in \ref{tab:empirical_data}. 
(A) The pbMutSel preferences are more correlated with the re-scaled average of the H1 and H3 deep mutational scanning preferences than the individual re-scaled H1 and H3 deep mutational scanning preferences are to each other (Pearson's $r$: 0.74 versus 0.52). 
(B) The ExpCM(H1+H3 avg)+$\Gamma\omega$ and pbMutSel models estimated similar branch lengths when fit to the entire HA tree. 
Points denote branch lengths between all pairs of tips on the tree. 
Blue and orange denote branches that lead to the H1 and H3 deep mutational scanning reference sequences respectively. 
The \texttt{phydms} program implementing ExpCMs and the \texttt{PhyloBayes-MPI} program implementing pbMutSel models give branch lengths in different units, so to facilitate direct comparison between the models, we have normalized all branch lengths returned by each program by the length of the branches separating the earliest (A/South Carolina/1918) and latest (A/Solomon Islands/2006) seasonal human H1 sequences on the tree. 
}
\end{figure}

%% tables
\newpage
\begin{table}[t!]
\caption{\label{tab:empirical_data}
{\bf Fitting of substitution models to the HA phylogenetic tree.}
All ExpCMs describe the evolution of HA better than the GY94 models, as evaluated by the Akaike information criteria~\citep[$\Delta$AIC,][]{posada2004model}.
The models fit here are the same ones in~\ref{fig:empirical_trees}. 
The $\omega$ value for each of the $K=4$ bins is shown for the models with $\Gamma\omega$ rate variation. 
All ExpCMs fit a stringency parameter $>1$.
} 
     \begin{tabular}{cccccccccc}
        \hline
         Model & $\Delta$AIC & {\shortstack{Log\\ Likelihood}} & $\omega$ & {\shortstack{Stringency\\ parameter ($\beta$)}}\\ \hline
       	ExpCM  (H1+H3 avg) + $\Gamma\omega$  & 0 & -51083 & 0.19,  0.50,  0.91,  1.86 &  1.69\\
	ExpCM (H1+H3 avg)  &  1063 & -51616 & 0.14 & 1.77\\
	ExpCM  (H1) + $\Gamma\omega$  & 1321 & -51744  & 0.12 ,  0.42,  0.89,  2.13 & 1.11\\
	ExpCM (H3) + $\Gamma\omega$ & 1777 & -51972 & 0.10,  0.36,  0.76,  1.84 & 1.28\\
	ExpCM (H1) & 2670 & -52419 &  0.12 & 1.21\\
	ExpCM (H3) &  3377 & -52773 & 0.12 & 1.43\\
	GY94 + $\Gamma\omega$  & 4817 & -53487 & 0.00,  0.03,  0.08,  0.24 & - \\
	GY94 & 7892 & -55025  & 0.07 & -\\
      \end{tabular}
\end{table}

\clearpage
\newpage
\setcounter{page}{1}
\renewcommand{\thepage}{S\arabic{page}}
\section*{Supplemental Information}

\begin{suppfig}[H]
\centerline{\includegraphics[width=\textwidth]{suppfig1.pdf}}
\caption{\label{suppfig:prefs_doud}
\textbf{H1 HA amino-acid preferences measured by deep mutational scanning.}
Each column represents a site in the HA protein, and the height of each letter is proportional to the preference for the amino acid measured by~\citet{doud2016accurate} and then re-scaled by the stringency parameter in \ref{tab:empirical_data}. 
The plot only shows sites that are alignable between the H1 and H3 HAs, and these alignable sites are numbered sequentially starting from 1.
The conversion between the numbering scheme in this figure and sequential numbering of the H1 HA reference sequence is in \ref{suppfile:WSN_Perth_map}. 
}
\end{suppfig}
\clearpage 


\begin{suppfig}[H]
\centerline{\includegraphics[width=\textwidth]{suppfig2.pdf}}
\caption{\label{suppfig:prefs_lee}
\textbf{H3 HA amino-acid preferences measured by deep mutational scanning.}
Similar to \ref{suppfig:prefs_doud} but shows the re-scaled preferences for the H3 HA as measured by \citet{lee2018deep}.
 }
\end{suppfig}
\clearpage 

\begin{suppfig}[H]
\centerline{\includegraphics[width=\textwidth]{suppfig3.pdf}}
\caption{\label{suppfig:prefs_average}
\textbf{Average of H1 HA  and H3 HA amino-acid preferences measured by deep mutational scanning.}
Similar to \ref{suppfig:prefs_doud} but shows the re-scaled average of the preferences for the H1 and H3 HAs.
}
\end{suppfig}
\clearpage 

\begin{suppfig}[H]
\centerline{\includegraphics[width=\textwidth]{suppfig4.pdf}}
\caption{\label{suppfig:prefs_mutSel}
\textbf{Amino-acid preferences inferred by the pbMutSel model.}
Similar to \ref{suppfig:prefs_doud}, but shows the preferences inferred by fitting the pbMutSel model to the full HA tree.
}
\end{suppfig}
\clearpage 

\begin{suppfig}[H]
\centerline{\includegraphics[width=0.50\textwidth]{suppfig5.pdf}}
\caption{\label{suppfig:subalignments}
\textbf{Overall divergence for subtrees.}
We created two subalignments for each HA used in the deep mutational scanning experiments. 
The ``low divergence" alignments had $\ge$59\% amino-acid identity to either the H1 or H3 reference sequence. 
The ``intermediate divergence" alignments had $\ge$46\% amino-acid identity to the reference sequences.
}
\end{suppfig}
\clearpage 

\begin{supptable}[t!]
\caption{\label{supptab:tree_diameter}
{\bf Branch length extension as measured by tree diameter.}
We calculated the tree diameter, the distance between the two most divergent tips, for the trees in \ref{fig:empirical_trees}.
For each tree, the diameter is reported as a raw value and as a percentage of the GY94 model tree, the smallest of the eight trees. 
} 
     \begin{tabular}{cccccccccc}
        \hline
         Model & {\shortstack{Tree diameter\\ (average codon substitutions per site)}} & {\shortstack{Percentage of\\ GY94 tree diameter}} \\ \hline
         GY94 & 12.04 & 100\%\\
         ExpCM(H1) & 14.70 & 122\%\\
         ExpCM(H3) & 16.28 & 135\%\\
         ExpCM(H1+H3 avg) & 19.21 & 160\%\\
         GY94 + $\Gamma\omega$ & 19.15 & 159\%\\
         ExpCM(H1) + $\Gamma\omega$ & 24.75 & 206\% \\
         ExpCM(H3) + $\Gamma\omega$ & 25.03 & 208\% \\
         ExpCM(H1+H3 avg) + $\Gamma\omega$ & 30.78 & 256\% \\
      \end{tabular}
\end{supptable}
\clearpage 

\begin{supptable}[t!]
\caption{\label{supptab:fit_params}
{\bf Model parameters fit to a low divergence tree.}
We fit GY94 models and an ExpCM defined by H1 deep mutational scanning preferences to the ``low divergence from H1" tree in \ref{fig:compete}. 
We used these model parameters calculate the expected pairwise sequence identity in \ref{fig:decay} and simulate the sequences in \ref{fig:simulations}. 
} 
     \begin{tabular}{ccc}
        \hline
         Model & Parameters \\ \hline
         GY94 &  {\shortstack{$\kappa = 3.17$, $\omega = 0.10$, \\$\phi_{1,A} = 0.32$, $\phi_{1,C} = 0.14$, $\phi_{1,G} = 0.28$, \\$\phi_{2,A} = 0.38$, $\phi_{2,C} = 0.18$,  $\phi_{2G} = 0.20$,\\ $\phi_{3, A} = 0.36$, $\phi_{3,C} = 0.19$, $\phi_{3,G} = 0.21$}}\\ \hline
         GY94 + $\Gamma\omega$ & {\shortstack{$\alpha_\omega = 0.51$, $\beta_\omega = 3.92$, $\kappa = 3.49$, \\$\phi_{1,A} = 0.32$, $\phi_{1,C} = 0.14$, $\phi_{1,G} = 0.28$, \\$\phi_{2,A} = 0.38$, $\phi_{2,C} = 0.18$, $\phi_{2,G} = 0.20$, \\$\phi_{3,A} = 0.36$, $\phi_{3,C} = 0.19$, $\phi_{3,G} = 0.21$}}\\ \hline
         ExpCM(H1) &  {\shortstack{$\beta = 1.56$, $\kappa = 3.64$, $\omega = 0.24$,\\ $\phi_A = 0.378$, $\phi_C = 0.17$, $\phi_G = 0.23$}}\\ \hline
      \end{tabular}
\end{supptable}
\clearpage 


%%%

\begin{suppfile}
\caption{
\label{suppfile:WSN_Perth_map}
List of alignable sites between H1 HA and H3 HA. 
This files provides a conversion between the numbering scheme we use in the paper (sequential numbering of just the alignable sites) to sequential numbering of the H1 HA reference sequence A/Wilson Smith/1933 and the H3 HA reference sequence A/Perth/2009. 
}
\end{suppfile}

\begin{suppfile}
\caption{
\label{suppfile:H1_prefs}
Amino acid preferences measured by the deep mutational scanning of the H1 HA strain A/WSN/1933~\citep{doud2016accurate}. 
This file only contains measurements for the alignable sites between H1 and H3 HAs. 
Conversion from this numbering scheme to sequential numbering of A/WSN/1933 is in \ref{suppfile:WSN_Perth_map}. 
}
\end{suppfile}

\begin{suppfile}
\caption{
\label{suppfile:H3_prefs}
Amino acid preferences measured by the deep mutational scanning of the H3 HA strain A/Perth/2009~\citep{lee2018deep}. 
This file only contains measurements for the alignable sites between H1 and H3 HAs. 
Conversion from this numbering scheme to sequential numbering of A/Perth/2009 is in in \ref{suppfile:WSN_Perth_map}. 
}
\end{suppfile}

\begin{suppfile}
\caption{
\label{suppfile:alignment_high}
The HA sequences for the full HA tree. 
The sequences in this alignment contain only the alignable sites between H1 and H3 HAs.
Conversion from this numbering scheme to sequential numbering of A/Perth/2009 is in in \ref{suppfile:WSN_Perth_map}. 
}
\end{suppfile}

\end{document}
